// igniteui-angular-wrappers/esm2015/lib/ighierarchicalgrid/ighierarchicalgrid.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Renderer, IterableDiffers, KeyValueDiffers, ChangeDetectorRef, ElementRef, ChangeDetectionStrategy } from "@angular/core";
import { IgGridBase } from "../iggrid/iggridbase";
export class IgHierarchicalGridComponent extends IgGridBase {
    /**
     * @param {?} el
     * @param {?} renderer
     * @param {?} differs
     * @param {?} kvalDiffers
     * @param {?} cdr
     */
    constructor(el, renderer, differs, kvalDiffers, cdr) { super(el, renderer, differs, kvalDiffers, cdr); }
    /**
     * @param {?} id
     * @return {?}
     */
    deleteRow(id) {
        /** @type {?} */
        var element = jQuery(this._el);
        /** @type {?} */
        var tr = element.find("tr[data-id='" + id + "']");
        /** @type {?} */
        var childContainer = tr.next("tr[data-container]");
        if (tr.length > 0) {
            tr.remove();
            childContainer.remove();
            element.data("igGrid").dataSource.deleteRow(id, true);
            element.data("igGrid").dataSource._removeTransactionsByRecordId(id);
        }
    }
    /**
     * @param {?} rec
     * @param {?} currValue
     * @param {?} key
     * @return {?}
     */
    updateRow(rec, currValue, key) {
        /** @type {?} */
        const element = jQuery(this._el);
        /** @type {?} */
        const childrenDataProperty = this["childrenDataProperty"] || this.options.childrenDataProperty;
        /** @type {?} */
        var childGrid = element.data(this._widgetName).allChildrenWidgets().filter((/**
         * @param {?} indx
         * @return {?}
         */
        function (indx) {
            /** @type {?} */
            var parentRow = jQuery(this.element).closest('tr[data-container]').prev();
            /** @type {?} */
            var parentGridPK = parentRow.closest(".ui-iggrid-table").data("igGrid").options.primaryKey;
            return (childrenDataProperty === key ||
                parentRow.next("[data-container]").find("table[role='grid']").attr("id").contains("_" + key + "_"))
                && parentRow.attr("data-id") == rec[parentGridPK];
        }));
        if (childGrid.length > 0) {
            jQuery(childGrid).each((/**
             * @return {?}
             */
            function () {
                this.dataBind();
            }));
        }
        else {
            super.updateRow(rec, currValue, key);
        }
    }
    /**
     * @return {?}
     */
    markForCheck() {
        super.markForCheck();
        /** @type {?} */
        const element = jQuery(this._el);
        /** @type {?} */
        const childrenDataProperty = this["childrenDataProperty"] || this.options.childrenDataProperty;
        /** @type {?} */
        var childGrids = element.data(this._widgetName).allChildrenWidgets();
        for (var i = 0; i < childGrids.length; i++) {
            childGrids[i].dataBind();
        }
    }
    /**
     * Data binds the hierarchical grid. No child grids will be created or rendered by default, unless there is initialExpandDepth >= 0 set.
     * @return {?}
     */
    /* istanbul ignore next */
    dataBind() { return; }
    ;
    /**
     * Returns the element of the root grid (igGrid)
     * @return {?}
     */
    /* istanbul ignore next */
    root() { return; }
    ;
    /**
     * Returns the widget object of the root grid (igGrid)
     * @return {?}
     */
    /* istanbul ignore next */
    rootWidget() { return; }
    ;
    /**
     * Returns a flat list of all child grid elements (recursive)
     * @return {?}
     */
    /* istanbul ignore next */
    allChildren() { return; }
    ;
    /**
     * Expands or collapses (toggles) a parent row
     * 				Note: This method is asynchronous which means that it returns immediately and any subsequent code will execute in parallel. This may lead to runtime errors. To avoid them put the subsequent code in the callback parameter provided by the method.
     *
     * @param {?} element     accepts a dom element, or a jquery wrapped dom element that should be a TR and should specify a parent row
     * @param {?=} callback     Specifies a custom function to be called when parent row is toggled(optional). Takes 2 arguments - first is hierarchical grid object, second is the row element that was toggled
     * @return {?}
     */
    /* istanbul ignore next */
    toggle(element, callback) { return; }
    ;
    /**
     * Expands (toggles) a parent row
     * 				Note: This method is asynchronous which means that it returns immediately and any subsequent code will execute in parallel. This may lead to runtime errors. To avoid them put the subsequent code in the callback parameter provided by the method.
     *
     * @param {?} id     accepts a dom element, or a jquery wrapped dom element that should be a TR and should specify a parent row
     * @param {?=} callback     Specifies a custom function to be called when parent row is expanded(optional). Takes 2 arguments first is hierarchical grid object, second is the row element that was expanded
     * @return {?}
     */
    /* istanbul ignore next */
    expand(id, callback) { return; }
    ;
    /**
     * Collapses a parent row
     * 				Note: This method is asynchronous which means that it returns immediately and any subsequent code will execute in parallel. This may lead to runtime errors. To avoid them put the subsequent code in the callback parameter provided by the method.
     *
     * @param {?} id     accepts a dom element, or a jquery wrapped dom element that should be a TR and should specify a parent row
     * @param {?=} callback     Specifies a custom function to be called when parent row is expanded(optional). Takes 2 arguments - first is hierarchical grid object, second is the row element that was collapsed
     * @return {?}
     */
    /* istanbul ignore next */
    collapse(id, callback) { return; }
    ;
    /**
     * Checks if a parent row is currently collapsed
     *
     * @param {?} element     accepts a dom element, or a jquery wrapped dom element that should be a TR and should specify a parent row
     * @return {?}
     */
    /* istanbul ignore next */
    collapsed(element) { return; }
    ;
    /**
     * Checks if a parent row is populated with data
     *
     * @param {?} element     accepts a dom element, or a jquery wrapped dom element that should be a TR and should specify a parent row
     * @return {?}
     */
    /* istanbul ignore next */
    populated(element) { return; }
    ;
    /**
     * Commits pending transactions to the client data source for main and all child grids.
     * @return {?}
     */
    /* istanbul ignore next */
    commit() { return; }
    ;
    /**
     * Clears the transaction log (delegates to igDataSource). Note that this does not update the UI. In case the UI must be updated, set the second parameter "updateUI" to true, which will trigger a call to dataBind() to re-render the contents.
     *
     * @param {?=} rebind     Whether to perform a rebind.
     * @return {?}
     */
    /* istanbul ignore next */
    rollback(rebind) { return; }
    ;
    /**
     * Posts to the settings.updateUrl using $.ajax, by serializing the changes as url params
     *
     * @param {?} success    Specifies a custom function to be called when AJAX request to the updateUrl option succeeds(optional)
     * @param {?} error    Specifies a custom function to be called when AJAX request to the updateUrl option fails(optional)
     * @return {?}
     */
    /* istanbul ignore next */
    saveChanges(success, error) { return; }
    ;
    /**
     * Destroys the hierarchical grid by recursively destroying all child grids
     * @return {?}
     */
    /* istanbul ignore next */
    destroy() { return; }
    ;
}
IgHierarchicalGridComponent.decorators = [
    { type: Component, args: [{
                changeDetection: ChangeDetectionStrategy.OnPush,
                selector: "ig-hierarchical-grid",
                template: "<ng-content></ng-content>",
                inputs: ["widgetId", "options", "changeDetectionInterval", "disabled", "create", "initialDataBindDepth", "initialExpandDepth", "odata", "rest", "maxDataBindDepth", "defaultChildrenDataProperty", "autoGenerateLayouts", "expandCollapseAnimations", "expandColWidth", "pathSeparator", "animationDuration", "expandTooltip", "collapseTooltip", "columnLayouts", "width", "height", "autoAdjustHeight", "avgRowHeight", "avgColumnWidth", "defaultColumnWidth", "autoGenerateColumns", "virtualization", "virtualizationMode", "requiresDataBinding", "rowVirtualization", "columnVirtualization", "virtualizationMouseWheelStep", "adjustVirtualHeights", "templatingEngine", "columns", "dataSource", "dataSourceUrl", "dataSourceType", "responseDataKey", "responseTotalRecCountKey", "requestType", "responseContentType", "showHeader", "showFooter", "fixedHeaders", "fixedFooters", "caption", "features", "tabIndex", "localSchemaTransform", "primaryKey", "serializeTransactionLog", "autoCommit", "aggregateTransactions", "autoFormat", "renderCheckboxes", "updateUrl", "restSettings", "alternateRowStyles", "autofitLastColumn", "enableHoverStyles", "enableUTCDates", "mergeUnboundColumns", "jsonpRequest", "enableResizeContainerCheck", "featureChooserIconDisplay", "scrollSettings"],
                outputs: ["rowExpanding", "rowExpanded", "rowCollapsing", "rowCollapsed", "childrenPopulating", "childrenPopulated", "childGridRendered", "childGridCreating", "childGridCreated", "cellClick", "cellRightClick", "dataBinding", "dataBound", "rendering", "rendered", "dataRendering", "dataRendered", "headerRendering", "headerRendered", "footerRendering", "footerRendered", "headerCellRendered", "rowsRendering", "rowsRendered", "schemaGenerated", "columnsCollectionModified", "requestError", "created", "destroyed"]
            }] }
];
/** @nocollapse */
IgHierarchicalGridComponent.ctorParameters = () => [
    { type: ElementRef },
    { type: Renderer },
    { type: IterableDiffers },
    { type: KeyValueDiffers },
    { type: ChangeDetectorRef }
];
