// igniteui-angular-wrappers/esm5/lib/iggrid/feature.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { EventEmitter } from "@angular/core";
/**
 * @template Model
 */
var /**
 * @template Model
 */
Feature = /** @class */ (function () {
    function Feature(el) {
        this._settings = {};
        this._el = el;
        this.name = this.normalizeName(el.nativeElement.nodeName.toLowerCase());
        this.featureName = "igGrid" + this.name;
        for (var propt in jQuery.ui["igGrid" + this.name].prototype.events) {
            this[propt] = new EventEmitter();
        }
    }
    /**
     * @return {?}
     */
    Feature.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        /** @type {?} */
        var self = this;
        this.initSettings = jQuery.extend(true, {}, this);
        /** @type {?} */
        var evtName;
        this._events = new Map();
        /** @type {?} */
        var grid = jQuery(this._el.nativeElement).closest("ig-grid").find("table");
        //event binding for features
        for (var propt in jQuery.ui[this.featureName].prototype.events) {
            evtName = this.featureName.toLowerCase() + propt.toLowerCase();
            this._events[evtName] = propt;
            jQuery(grid).on(evtName, (/**
             * @param {?} evt
             * @param {?} ui
             * @return {?}
             */
            function (evt, ui) {
                self[self._events[evt.type]].emit({ event: evt, ui: ui });
            }));
        }
        for (var setting in jQuery.ui[this.featureName].prototype.options) {
            Object.defineProperty(self, setting, {
                set: self.createFeatureSetter(setting),
                get: self.createFeatureGetter(setting),
                enumerable: true,
                configurable: true
            });
        }
        /** @type {?} */
        var propNames = Object.getOwnPropertyNames(jQuery.ui[this.featureName].prototype);
        for (var i = 0; i < propNames.length; i++) {
            /** @type {?} */
            var name = propNames[i];
            if (name.indexOf("_") !== 0 && typeof jQuery.ui[this.featureName].prototype[name] === "function") {
                Object.defineProperty(self, name, {
                    get: self.createMethodGetter(name)
                });
            }
        }
    };
    /**
     * @param {?} name
     * @return {?}
     */
    Feature.prototype.createFeatureSetter = /**
     * @param {?} name
     * @return {?}
     */
    function (name) {
        return (/**
         * @param {?} value
         * @return {?}
         */
        function (value) {
            /** @type {?} */
            var grid = jQuery(this._el.nativeElement).closest("ig-grid").find("table[role='grid']");
            this._settings[name] = value;
            if (jQuery.ui[this.featureName] &&
                jQuery.ui[this.featureName].prototype.options &&
                jQuery.ui[this.featureName].prototype.options.hasOwnProperty(name) &&
                grid.data(this.featureName)) {
                grid[this.featureName]("option", name, value);
            }
        });
    };
    /**
     * @param {?} name
     * @return {?}
     */
    Feature.prototype.createFeatureGetter = /**
     * @param {?} name
     * @return {?}
     */
    function (name) {
        return (/**
         * @return {?}
         */
        function () {
            return this._settings[name];
        });
    };
    /**
     * @param {?} name
     * @return {?}
     */
    Feature.prototype.createMethodGetter = /**
     * @param {?} name
     * @return {?}
     */
    function (name) {
        return (/**
         * @return {?}
         */
        function () {
            /** @type {?} */
            var grid = jQuery(this._el.nativeElement).closest("ig-grid").find("table[role='grid']");
            if (grid.length === 0) {
                grid = jQuery(this._el.nativeElement).closest("ig-hierarchical-grid").find("table[role='grid']");
            }
            /** @type {?} */
            var args = [];
            /** @type {?} */
            var feature = grid.data(this.featureName);
            return jQuery.proxy(feature[name], feature);
        });
    };
    /**
     * @param {?} str
     * @return {?}
     */
    Feature.prototype.normalizeName = /**
     * @param {?} str
     * @return {?}
     */
    function (str) {
        //convert hyphen to camelCase
        /** @type {?} */
        var name = str.replace(/-([a-z])/g, (/**
         * @param {?} group
         * @return {?}
         */
        function (group) {
            return group[1].toUpperCase();
        }));
        return name.charAt(0).toUpperCase() + name.slice(1);
    };
    return Feature;
}());
/**
 * @template Model
 */
export { Feature };
if (false) {
    /** @type {?} */
    Feature.prototype._el;
    /** @type {?} */
    Feature.prototype._settings;
    /** @type {?} */
    Feature.prototype.initSettings;
    /** @type {?} */
    Feature.prototype.name;
    /**
     * @type {?}
     * @private
     */
    Feature.prototype._events;
    /**
     * @type {?}
     * @private
     */
    Feature.prototype.featureName;
}
