// igniteui-angular-wrappers/esm5/lib/iggrid/iggridfeatures/iggridsummaries.directive.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { ElementRef, Directive } from "@angular/core";
import { Feature } from "../feature";
var IgGridSummariesFeature = /** @class */ (function (_super) {
    tslib_1.__extends(IgGridSummariesFeature, _super);
    function IgGridSummariesFeature(el) {
        return _super.call(this, el) || this;
    }
    /* istanbul ignore next */
    /* istanbul ignore next */
    /**
     * @return {?}
     */
    IgGridSummariesFeature.prototype.destroy = /* istanbul ignore next */
    /**
     * @return {?}
     */
    function () { return; };
    ;
    /**
     * Returns whether summaries rows are hidden
     */
    /* istanbul ignore next */
    /**
     * Returns whether summaries rows are hidden
     * @return {?}
     */
    /* istanbul ignore next */
    IgGridSummariesFeature.prototype.isSummariesRowsHidden = /**
     * Returns whether summaries rows are hidden
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    /**
     * Calculate summaries
     */
    /* istanbul ignore next */
    /**
     * Calculate summaries
     * @return {?}
     */
    /* istanbul ignore next */
    IgGridSummariesFeature.prototype.calculateSummaries = /**
     * Calculate summaries
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    /**
     * Remove all summaries dropdown buttons.
     */
    /* istanbul ignore next */
    /**
     * Remove all summaries dropdown buttons.
     * @return {?}
     */
    /* istanbul ignore next */
    IgGridSummariesFeature.prototype.clearAllFooterIcons = /**
     * Remove all summaries dropdown buttons.
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    /**
     * Toggle drop down
     *
     * @param columnKey    toggle drop down for the column with the specified key
     * @param event    event object. Its data should contain current columnKey, isAnimating, buttonId
     */
    /* istanbul ignore next */
    /**
     * Toggle drop down
     *
     * @param {?} columnKey    toggle drop down for the column with the specified key
     * @param {?} event    event object. Its data should contain current columnKey, isAnimating, buttonId
     * @return {?}
     */
    /* istanbul ignore next */
    IgGridSummariesFeature.prototype.toggleDropDown = /**
     * Toggle drop down
     *
     * @param {?} columnKey    toggle drop down for the column with the specified key
     * @param {?} event    event object. Its data should contain current columnKey, isAnimating, buttonId
     * @return {?}
     */
    /* istanbul ignore next */
    function (columnKey, event) { return; };
    ;
    /**
     * Show/Hide dialog
     *
     * @param $dialog     jQuery object representation of dropdown div element
     */
    /* istanbul ignore next */
    /**
     * Show/Hide dialog
     *
     * @param {?} $dialog     jQuery object representation of dropdown div element
     * @return {?}
     */
    /* istanbul ignore next */
    IgGridSummariesFeature.prototype.showHideDialog = /**
     * Show/Hide dialog
     *
     * @param {?} $dialog     jQuery object representation of dropdown div element
     * @return {?}
     */
    /* istanbul ignore next */
    function ($dialog) { return; };
    ;
    /**
     * Toggle summaries rows
     *
     * @param isToShow    Specifies whether to show or not summaries
     * @param isInternalCall    Optional parameter.Specifies whether this function is called internally by the widget.
     */
    /* istanbul ignore next */
    /**
     * Toggle summaries rows
     *
     * @param {?} isToShow    Specifies whether to show or not summaries
     * @param {?} isInternalCall    Optional parameter.Specifies whether this function is called internally by the widget.
     * @return {?}
     */
    /* istanbul ignore next */
    IgGridSummariesFeature.prototype.toggleSummariesRows = /**
     * Toggle summaries rows
     *
     * @param {?} isToShow    Specifies whether to show or not summaries
     * @param {?} isInternalCall    Optional parameter.Specifies whether this function is called internally by the widget.
     * @return {?}
     */
    /* istanbul ignore next */
    function (isToShow, isInternalCall) { return; };
    ;
    /**
     * Toggles the checkstate of a checkbox if checkboxMode is not set to off, otherwise does nothing.
     *
     * @param $checkbox     Specifies the jQuery object of the checkbox.
     */
    /* istanbul ignore next */
    /**
     * Toggles the checkstate of a checkbox if checkboxMode is not set to off, otherwise does nothing.
     *
     * @param {?} $checkbox     Specifies the jQuery object of the checkbox.
     * @return {?}
     */
    /* istanbul ignore next */
    IgGridSummariesFeature.prototype.toggleCheckstate = /**
     * Toggles the checkstate of a checkbox if checkboxMode is not set to off, otherwise does nothing.
     *
     * @param {?} $checkbox     Specifies the jQuery object of the checkbox.
     * @return {?}
     */
    /* istanbul ignore next */
    function ($checkbox) { return; };
    ;
    /**
     * Select/Unselect specified checkbox
     *
     * @param $checkbox     Specifies the jQuery object for checkbox
     * @param isToSelect     Specify whether to select or not checkbox
     */
    /* istanbul ignore next */
    /**
     * Select/Unselect specified checkbox
     *
     * @param {?} $checkbox     Specifies the jQuery object for checkbox
     * @param {?} isToSelect     Specify whether to select or not checkbox
     * @return {?}
     */
    /* istanbul ignore next */
    IgGridSummariesFeature.prototype.selectCheckBox = /**
     * Select/Unselect specified checkbox
     *
     * @param {?} $checkbox     Specifies the jQuery object for checkbox
     * @param {?} isToSelect     Specify whether to select or not checkbox
     * @return {?}
     */
    /* istanbul ignore next */
    function ($checkbox, isToSelect) { return; };
    ;
    /**
     * Summary calculate the whole data for the specified column key, columnMethods and dataType (used when datasource is remote and dataType is date)
     *
     * @param ck    ColumnKey
     * @param columnMethods    Array of column methods objects
     * @param data    Object which represents result
 represents dataType for the current column
    * @param dataType
    */
    /* istanbul ignore next */
    /**
     * Summary calculate the whole data for the specified column key, columnMethods and dataType (used when datasource is remote and dataType is date)
     *
     * @param {?} ck    ColumnKey
     * @param {?} columnMethods    Array of column methods objects
     * @param {?} data    Object which represents result
     * represents dataType for the current column
     * @param {?} dataType
     * @return {?}
     */
    /* istanbul ignore next */
    IgGridSummariesFeature.prototype.calculateSummaryColumn = /**
     * Summary calculate the whole data for the specified column key, columnMethods and dataType (used when datasource is remote and dataType is date)
     *
     * @param {?} ck    ColumnKey
     * @param {?} columnMethods    Array of column methods objects
     * @param {?} data    Object which represents result
     * represents dataType for the current column
     * @param {?} dataType
     * @return {?}
     */
    /* istanbul ignore next */
    function (ck, columnMethods, data, dataType) { return; };
    ;
    /**
     * Return a JQUERY object which holds all summaries for all columns
     */
    /* istanbul ignore next */
    /**
     * Return a JQUERY object which holds all summaries for all columns
     * @return {?}
     */
    /* istanbul ignore next */
    IgGridSummariesFeature.prototype.summaryCollection = /**
     * Return a JQUERY object which holds all summaries for all columns
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    /**
     * Return a JQUERY object which holds all summaries for column with the specified column key
     *
     * @param columnKey
     */
    /* istanbul ignore next */
    /**
     * Return a JQUERY object which holds all summaries for column with the specified column key
     *
     * @param {?} columnKey
     * @return {?}
     */
    /* istanbul ignore next */
    IgGridSummariesFeature.prototype.summariesFor = /**
     * Return a JQUERY object which holds all summaries for column with the specified column key
     *
     * @param {?} columnKey
     * @return {?}
     */
    /* istanbul ignore next */
    function (columnKey) { return; };
    ;
    IgGridSummariesFeature.decorators = [
        { type: Directive, args: [{
                    selector: 'summaries',
                    inputs: ["disabled", "create", "type", "dialogButtonOKText", "dialogButtonCancelText", "calculateRenderMode", "featureChooserText", "featureChooserTextHide", "compactRenderingMode", "defaultDecimalDisplay", "showSummariesButton", "summariesResponseKey", "summaryExprUrlKey", "callee", "dropDownHeight", "dropDownWidth", "showDropDownButton", "summaryExecution", "dropDownDialogAnimationDuration", "emptyCellText", "summariesHeaderButtonTooltip", "resultTemplate", "isGridFormatter", "renderSummaryCellFunc", "columnSettings", "inherit"],
                    outputs: ["dropDownOpening", "dropDownOpened", "dropDownClosing", "dropDownClosed", "summariesCalculating", "summariesCalculated", "summariesMethodSelectionChanged", "summariesToggling", "summariesToggled", "dropDownOKClicked", "dropDownCancelClicked"]
                },] }
    ];
    /** @nocollapse */
    IgGridSummariesFeature.ctorParameters = function () { return [
        { type: ElementRef }
    ]; };
    return IgGridSummariesFeature;
}(Feature));
export { IgGridSummariesFeature };
