// igniteui-angular-wrappers/esm5/lib/iggrid/iggridfeatures/iggridhiding.directive.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { ElementRef, Directive } from "@angular/core";
import { Feature } from "../feature";
var IgGridHidingFeature = /** @class */ (function (_super) {
    tslib_1.__extends(IgGridHidingFeature, _super);
    function IgGridHidingFeature(el) {
        return _super.call(this, el) || this;
    }
    /**
     * Destroys the hiding widget
     */
    /* istanbul ignore next */
    /**
     * Destroys the hiding widget
     * @return {?}
     */
    /* istanbul ignore next */
    IgGridHidingFeature.prototype.destroy = /**
     * Destroys the hiding widget
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    /**
     * Shows the Column Chooser dialog. If it is visible the method does nothing.
     */
    /* istanbul ignore next */
    /**
     * Shows the Column Chooser dialog. If it is visible the method does nothing.
     * @return {?}
     */
    /* istanbul ignore next */
    IgGridHidingFeature.prototype.showColumnChooser = /**
     * Shows the Column Chooser dialog. If it is visible the method does nothing.
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    /**
     * Hides the Column Chooser dialog. If it is not visible the method does nothing.
     */
    /* istanbul ignore next */
    /**
     * Hides the Column Chooser dialog. If it is not visible the method does nothing.
     * @return {?}
     */
    /* istanbul ignore next */
    IgGridHidingFeature.prototype.hideColumnChooser = /**
     * Hides the Column Chooser dialog. If it is not visible the method does nothing.
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    /**
     * Shows a hidden column. If the column is not hidden the method does nothing.
     * 			Note: This method is asynchronous which means that it returns immediately and any subsequent code will execute in parallel. This may lead to runtime errors. To avoid them put the subsequent code in the callback parameter provided by the method.
     *
     * @param column    An identifier for the column. If a number is provided it will be used as a column index else if a strings is provided it will be used as a column key.
     * @param isMultiColumnHeader    If it is true then the column is of type multicolumnheader. An identifier for the column should be of type string.
     * @param callback    Specifies a custom function to be called when the column(s) is shown(optional)
     */
    /* istanbul ignore next */
    /**
     * Shows a hidden column. If the column is not hidden the method does nothing.
     * 			Note: This method is asynchronous which means that it returns immediately and any subsequent code will execute in parallel. This may lead to runtime errors. To avoid them put the subsequent code in the callback parameter provided by the method.
     *
     * @param {?} column    An identifier for the column. If a number is provided it will be used as a column index else if a strings is provided it will be used as a column key.
     * @param {?=} isMultiColumnHeader    If it is true then the column is of type multicolumnheader. An identifier for the column should be of type string.
     * @param {?=} callback    Specifies a custom function to be called when the column(s) is shown(optional)
     * @return {?}
     */
    /* istanbul ignore next */
    IgGridHidingFeature.prototype.showColumn = /**
     * Shows a hidden column. If the column is not hidden the method does nothing.
     * 			Note: This method is asynchronous which means that it returns immediately and any subsequent code will execute in parallel. This may lead to runtime errors. To avoid them put the subsequent code in the callback parameter provided by the method.
     *
     * @param {?} column    An identifier for the column. If a number is provided it will be used as a column index else if a strings is provided it will be used as a column key.
     * @param {?=} isMultiColumnHeader    If it is true then the column is of type multicolumnheader. An identifier for the column should be of type string.
     * @param {?=} callback    Specifies a custom function to be called when the column(s) is shown(optional)
     * @return {?}
     */
    /* istanbul ignore next */
    function (column, isMultiColumnHeader, callback) { return; };
    ;
    /**
     * Hides a visible column. If the column is hidden the method does nothing.
     * 			Note: This method is asynchronous which means that it returns immediately and any subsequent code will execute in parallel. This may lead to runtime errors. To avoid them put the subsequent code in the callback parameter provided by the method.
     *
     * @param column    An identifier for the column. If a number is provided it will be used as a column index else if a strings is provided it will be used as a column key.
     * @param isMultiColumnHeader    If it is true then the column is of type multicolumnheader. An identifier for the column should be of type string.
     * @param callback    Specifies a custom function to be called when the column is hidden(optional)
     */
    /* istanbul ignore next */
    /**
     * Hides a visible column. If the column is hidden the method does nothing.
     * 			Note: This method is asynchronous which means that it returns immediately and any subsequent code will execute in parallel. This may lead to runtime errors. To avoid them put the subsequent code in the callback parameter provided by the method.
     *
     * @param {?} column    An identifier for the column. If a number is provided it will be used as a column index else if a strings is provided it will be used as a column key.
     * @param {?=} isMultiColumnHeader    If it is true then the column is of type multicolumnheader. An identifier for the column should be of type string.
     * @param {?=} callback    Specifies a custom function to be called when the column is hidden(optional)
     * @return {?}
     */
    /* istanbul ignore next */
    IgGridHidingFeature.prototype.hideColumn = /**
     * Hides a visible column. If the column is hidden the method does nothing.
     * 			Note: This method is asynchronous which means that it returns immediately and any subsequent code will execute in parallel. This may lead to runtime errors. To avoid them put the subsequent code in the callback parameter provided by the method.
     *
     * @param {?} column    An identifier for the column. If a number is provided it will be used as a column index else if a strings is provided it will be used as a column key.
     * @param {?=} isMultiColumnHeader    If it is true then the column is of type multicolumnheader. An identifier for the column should be of type string.
     * @param {?=} callback    Specifies a custom function to be called when the column is hidden(optional)
     * @return {?}
     */
    /* istanbul ignore next */
    function (column, isMultiColumnHeader, callback) { return; };
    ;
    /**
     * Hides visible columns specified by the array. If the column is hidden the method does nothing.
     * 			Note: This method is asynchronous which means that it returns immediately and any subsequent code will execute in parallel. This may lead to runtime errors. To avoid them put the subsequent code in the callback parameter provided by the method.
     *
     * @param columns    An array of identifiers for the columns. If a number is provided it will be used as a column index else if a strings is provided it will be used as a column key.
     * @param callback    Specifies a custom function to be called when all columns are hidden(optional)
     */
    /* istanbul ignore next */
    /**
     * Hides visible columns specified by the array. If the column is hidden the method does nothing.
     * 			Note: This method is asynchronous which means that it returns immediately and any subsequent code will execute in parallel. This may lead to runtime errors. To avoid them put the subsequent code in the callback parameter provided by the method.
     *
     * @param {?} columns    An array of identifiers for the columns. If a number is provided it will be used as a column index else if a strings is provided it will be used as a column key.
     * @param {?=} callback    Specifies a custom function to be called when all columns are hidden(optional)
     * @return {?}
     */
    /* istanbul ignore next */
    IgGridHidingFeature.prototype.hideMultiColumns = /**
     * Hides visible columns specified by the array. If the column is hidden the method does nothing.
     * 			Note: This method is asynchronous which means that it returns immediately and any subsequent code will execute in parallel. This may lead to runtime errors. To avoid them put the subsequent code in the callback parameter provided by the method.
     *
     * @param {?} columns    An array of identifiers for the columns. If a number is provided it will be used as a column index else if a strings is provided it will be used as a column key.
     * @param {?=} callback    Specifies a custom function to be called when all columns are hidden(optional)
     * @return {?}
     */
    /* istanbul ignore next */
    function (columns, callback) { return; };
    ;
    /**
     * Show visible columns specified by the array. If the column is shown the method does nothing.
     * 			Note: This method is asynchronous which means that it returns immediately and any subsequent code will execute in parallel. This may lead to runtime errors. To avoid them put the subsequent code in the callback parameter provided by the method.
     *
     * @param columns    An array of identifiers for the columns. If a number is provided it will be used as a column index else if a strings is provided it will be used as a column key.
     * @param callback    Specifies a custom function to be called when all columns are shown(optional)
     */
    /* istanbul ignore next */
    /**
     * Show visible columns specified by the array. If the column is shown the method does nothing.
     * 			Note: This method is asynchronous which means that it returns immediately and any subsequent code will execute in parallel. This may lead to runtime errors. To avoid them put the subsequent code in the callback parameter provided by the method.
     *
     * @param {?} columns    An array of identifiers for the columns. If a number is provided it will be used as a column index else if a strings is provided it will be used as a column key.
     * @param {?=} callback    Specifies a custom function to be called when all columns are shown(optional)
     * @return {?}
     */
    /* istanbul ignore next */
    IgGridHidingFeature.prototype.showMultiColumns = /**
     * Show visible columns specified by the array. If the column is shown the method does nothing.
     * 			Note: This method is asynchronous which means that it returns immediately and any subsequent code will execute in parallel. This may lead to runtime errors. To avoid them put the subsequent code in the callback parameter provided by the method.
     *
     * @param {?} columns    An array of identifiers for the columns. If a number is provided it will be used as a column index else if a strings is provided it will be used as a column key.
     * @param {?=} callback    Specifies a custom function to be called when all columns are shown(optional)
     * @return {?}
     */
    /* istanbul ignore next */
    function (columns, callback) { return; };
    ;
    /**
     * Gets whether the reset button in the column chooser dialog is to be rendered or not.
     */
    /* istanbul ignore next */
    /**
     * Gets whether the reset button in the column chooser dialog is to be rendered or not.
     * @return {?}
     */
    /* istanbul ignore next */
    IgGridHidingFeature.prototype.isToRenderButtonReset = /**
     * Gets whether the reset button in the column chooser dialog is to be rendered or not.
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    /**
     * Reset hidden/shown column to initial state of dialog(when it is opened)
     */
    /* istanbul ignore next */
    /**
     * Reset hidden/shown column to initial state of dialog(when it is opened)
     * @return {?}
     */
    /* istanbul ignore next */
    IgGridHidingFeature.prototype.resetHidingColumnChooser = /**
     * Reset hidden/shown column to initial state of dialog(when it is opened)
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    /**
     * Renders the Reset button in the Column Chooser dialog.
     */
    /* istanbul ignore next */
    /**
     * Renders the Reset button in the Column Chooser dialog.
     * @return {?}
     */
    /* istanbul ignore next */
    IgGridHidingFeature.prototype.renderColumnChooserResetButton = /**
     * Renders the Reset button in the Column Chooser dialog.
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    /**
     * Remove Reset button in column chooser modal dialog
     */
    /* istanbul ignore next */
    /**
     * Remove Reset button in column chooser modal dialog
     * @return {?}
     */
    /* istanbul ignore next */
    IgGridHidingFeature.prototype.removeColumnChooserResetButton = /**
     * Remove Reset button in column chooser modal dialog
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    IgGridHidingFeature.decorators = [
        { type: Directive, args: [{
                    selector: 'hiding',
                    inputs: ["disabled", "create", "columnSettings", "hiddenColumnIndicatorHeaderWidth", "columnChooserContainment", "columnChooserWidth", "columnChooserHeight", "dropDownAnimationDuration", "columnChooserCaptionText", "columnChooserDisplayText", "hiddenColumnIndicatorTooltipText", "columnHideText", "columnChooserShowText", "columnChooserHideText", "columnChooserHideOnClick", "columnChooserResetButtonLabel", "columnChooserAnimationDuration", "columnChooserButtonApplyText", "columnChooserButtonCancelText", "dialogWidget", "inherit"],
                    outputs: ["columnHiding", "columnHidingRefused", "columnShowingRefused", "multiColumnHiding", "columnHidden", "columnShowing", "columnShown", "columnChooserOpening", "columnChooserOpened", "columnChooserMoving", "columnChooserClosing", "columnChooserClosed", "columnChooserContentsRendering", "columnChooserContentsRendered", "columnChooserButtonApplyClick", "columnChooserButtonResetClick"]
                },] }
    ];
    /** @nocollapse */
    IgGridHidingFeature.ctorParameters = function () { return [
        { type: ElementRef }
    ]; };
    return IgGridHidingFeature;
}(Feature));
export { IgGridHidingFeature };
