// igniteui-angular-wrappers/esm5/lib/iggrid/iggridfeatures/iggridsorting.directive.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { ElementRef, Directive } from "@angular/core";
import { Feature } from "../feature";
var IgGridSortingFeature = /** @class */ (function (_super) {
    tslib_1.__extends(IgGridSortingFeature, _super);
    function IgGridSortingFeature(el) {
        return _super.call(this, el) || this;
    }
    /**
     * Sorts the data in a grid column  and updates the UI.
     *
     * @param index     Column key (string) or index (number) - for multi-row grid only column key can be used. Specifies the column which we want to sort. If the mode is multiple, previous sorting states are not cleared.
     * @param direction     Specifies sorting direction (ascending or descending)
     * @param header
     */
    /* istanbul ignore next */
    /**
     * Sorts the data in a grid column  and updates the UI.
     *
     * @param {?} index     Column key (string) or index (number) - for multi-row grid only column key can be used. Specifies the column which we want to sort. If the mode is multiple, previous sorting states are not cleared.
     * @param {?} direction     Specifies sorting direction (ascending or descending)
     * @param {?} header
     * @return {?}
     */
    /* istanbul ignore next */
    IgGridSortingFeature.prototype.sortColumn = /**
     * Sorts the data in a grid column  and updates the UI.
     *
     * @param {?} index     Column key (string) or index (number) - for multi-row grid only column key can be used. Specifies the column which we want to sort. If the mode is multiple, previous sorting states are not cleared.
     * @param {?} direction     Specifies sorting direction (ascending or descending)
     * @param {?} header
     * @return {?}
     */
    /* istanbul ignore next */
    function (index, direction, header) { return; };
    ;
    /**
     * Sorts the data in grid columns and updates the UI.\
     */
    /* istanbul ignore next */
    /**
     * Sorts the data in grid columns and updates the UI.\
     * @return {?}
     */
    /* istanbul ignore next */
    IgGridSortingFeature.prototype.sortMultiple = /**
     * Sorts the data in grid columns and updates the UI.\
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    /**
     * Removes current sorting(for all sorted columns) and updates the UI.
     */
    /* istanbul ignore next */
    /**
     * Removes current sorting(for all sorted columns) and updates the UI.
     * @return {?}
     */
    /* istanbul ignore next */
    IgGridSortingFeature.prototype.clearSorting = /**
     * Removes current sorting(for all sorted columns) and updates the UI.
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    /**
     * Removes sorting for the grid column with the specified columnKey/columnIndex and updates the UI.
     *
     * @param index     Column key (string) or index (number) - for multi-row grid only column key can be used. Specifies the column for which we want to remove sorting. If the mode is multiple, previous sorting states are not cleared.
     * @param header     - if specified client events should be fired
     */
    /* istanbul ignore next */
    /**
     * Removes sorting for the grid column with the specified columnKey/columnIndex and updates the UI.
     *
     * @param {?} index     Column key (string) or index (number) - for multi-row grid only column key can be used. Specifies the column for which we want to remove sorting. If the mode is multiple, previous sorting states are not cleared.
     * @param {?} header     - if specified client events should be fired
     * @return {?}
     */
    /* istanbul ignore next */
    IgGridSortingFeature.prototype.unsortColumn = /**
     * Removes sorting for the grid column with the specified columnKey/columnIndex and updates the UI.
     *
     * @param {?} index     Column key (string) or index (number) - for multi-row grid only column key can be used. Specifies the column for which we want to remove sorting. If the mode is multiple, previous sorting states are not cleared.
     * @param {?} header     - if specified client events should be fired
     * @return {?}
     */
    /* istanbul ignore next */
    function (index, header) { return; };
    ;
    /**
     * Destroys the sorting feature. Unbinds events, removes added sorting elements, etc.
     */
    /* istanbul ignore next */
    /**
     * Destroys the sorting feature. Unbinds events, removes added sorting elements, etc.
     * @return {?}
     */
    /* istanbul ignore next */
    IgGridSortingFeature.prototype.destroy = /**
     * Destroys the sorting feature. Unbinds events, removes added sorting elements, etc.
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    /**
     * Opens the multiple sorting dialog.
     */
    /* istanbul ignore next */
    /**
     * Opens the multiple sorting dialog.
     * @return {?}
     */
    /* istanbul ignore next */
    IgGridSortingFeature.prototype.openMultipleSortingDialog = /**
     * Opens the multiple sorting dialog.
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    /**
     * Closes the multiple sorting dialog.
     */
    /* istanbul ignore next */
    /**
     * Closes the multiple sorting dialog.
     * @return {?}
     */
    /* istanbul ignore next */
    IgGridSortingFeature.prototype.closeMultipleSortingDialog = /**
     * Closes the multiple sorting dialog.
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    /**
     * Renders content of multiple sorting dialog - sorted and unsorted columns.
     *
     * @param isToCallEvents
     */
    /* istanbul ignore next */
    /**
     * Renders content of multiple sorting dialog - sorted and unsorted columns.
     *
     * @param {?} isToCallEvents
     * @return {?}
     */
    /* istanbul ignore next */
    IgGridSortingFeature.prototype.renderMultipleSortingDialogContent = /**
     * Renders content of multiple sorting dialog - sorted and unsorted columns.
     *
     * @param {?} isToCallEvents
     * @return {?}
     */
    /* istanbul ignore next */
    function (isToCallEvents) { return; };
    ;
    /**
     * Remove clear button for multiple sorting dialog
     */
    /* istanbul ignore next */
    /**
     * Remove clear button for multiple sorting dialog
     * @return {?}
     */
    /* istanbul ignore next */
    IgGridSortingFeature.prototype.removeDialogClearButton = /**
     * Remove clear button for multiple sorting dialog
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    IgGridSortingFeature.decorators = [
        { type: Directive, args: [{
                    selector: 'sorting',
                    inputs: ["disabled", "create", "type", "caseSensitive", "applySortedColumnCss", "sortUrlKey", "sortUrlKeyAscValue", "sortUrlKeyDescValue", "mode", "customSortFunction", "firstSortDirection", "sortedColumnTooltip", "modalDialogSortOnClick", "modalDialogSortByButtonText", "modalDialogResetButtonLabel", "modalDialogCaptionButtonDesc", "modalDialogCaptionButtonAsc", "modalDialogCaptionButtonUnsort", "modalDialogWidth", "modalDialogHeight", "modalDialogAnimationDuration", "featureChooserText", "unsortedColumnTooltip", "columnSettings", "modalDialogCaptionText", "modalDialogButtonApplyText", "modalDialogButtonCancelText", "featureChooserSortAsc", "featureChooserSortDesc", "persist", "sortingDialogContainment", "dialogWidget", "inherit"],
                    outputs: ["columnSorting", "columnSorted", "modalDialogOpening", "modalDialogOpened", "modalDialogMoving", "modalDialogClosing", "modalDialogClosed", "modalDialogContentsRendering", "modalDialogContentsRendered", "modalDialogSortingChanged", "modalDialogButtonUnsortClick", "modalDialogSortClick", "modalDialogButtonApplyClick", "modalDialogButtonResetClick"]
                },] }
    ];
    /** @nocollapse */
    IgGridSortingFeature.ctorParameters = function () { return [
        { type: ElementRef }
    ]; };
    return IgGridSortingFeature;
}(Feature));
export { IgGridSortingFeature };
