// igniteui-angular-wrappers/esm5/lib/iggrid/iggridfeatures/iggridfiltering.directive.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { ElementRef, Directive } from "@angular/core";
import { Feature } from "../feature";
var IgGridFilteringFeature = /** @class */ (function (_super) {
    tslib_1.__extends(IgGridFilteringFeature, _super);
    function IgGridFilteringFeature(el) {
        return _super.call(this, el) || this;
    }
    /**
     * Destroys the filtering widget - remove fitler row, unbinds events, returns the grid to its previous state.
     */
    /* istanbul ignore next */
    /**
     * Destroys the filtering widget - remove fitler row, unbinds events, returns the grid to its previous state.
     * @return {?}
     */
    /* istanbul ignore next */
    IgGridFilteringFeature.prototype.destroy = /**
     * Destroys the filtering widget - remove fitler row, unbinds events, returns the grid to its previous state.
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    /**
     * Returns the count of data records that match filtering conditions
     */
    /* istanbul ignore next */
    /**
     * Returns the count of data records that match filtering conditions
     * @return {?}
     */
    /* istanbul ignore next */
    IgGridFilteringFeature.prototype.getFilteringMatchesCount = /**
     * Returns the count of data records that match filtering conditions
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    /**
     * Toggle filter row when mode is simple or [advancedModeEditorsVisible](ui.iggridfiltering#options:advancedModeEditorsVisible) is true. Otherwise show/hide advanced dialog.
     *
     * @param event     Column key
     */
    /* istanbul ignore next */
    /**
     * Toggle filter row when mode is simple or [advancedModeEditorsVisible](ui.iggridfiltering#options:advancedModeEditorsVisible) is true. Otherwise show/hide advanced dialog.
     *
     * @param {?} event     Column key
     * @return {?}
     */
    /* istanbul ignore next */
    IgGridFilteringFeature.prototype.toggleFilterRowByFeatureChooser = /**
     * Toggle filter row when mode is simple or [advancedModeEditorsVisible](ui.iggridfiltering#options:advancedModeEditorsVisible) is true. Otherwise show/hide advanced dialog.
     *
     * @param {?} event     Column key
     * @return {?}
     */
    /* istanbul ignore next */
    function (event) { return; };
    ;
    /**
     * Applies filtering programmatically and updates the UI by default.
     *
     * @param expressions     An array of filtering expressions, each one having the format {fieldName: , expr: , cond: , logic: } where  fieldName is the key of the column, expr is the actual expression string with which we would like to filter, logic is 'AND' or 'OR', and cond is one of the following strings: "equals", "doesNotEqual", "contains", "doesNotContain", "greaterThan", "lessThan", "greaterThanOrEqualTo", "lessThanOrEqualTo", "true", "false", "null", "notNull", "empty", "notEmpty", "startsWith", "endsWith", "today", "yesterday", "on", "notOn", "thisMonth", "lastMonth", "nextMonth", "before", "after", "thisYear", "lastYear", "nextYear". The difference between the empty and null filtering conditions is that empty includes null, NaN, and undefined, as well as the empty string.
     * @param updateUI     specifies whether the filter row should be also updated once the grid is filtered
     * @param addedFromAdvanced
     */
    /* istanbul ignore next */
    /**
     * Applies filtering programmatically and updates the UI by default.
     *
     * @param {?} expressions     An array of filtering expressions, each one having the format {fieldName: , expr: , cond: , logic: } where  fieldName is the key of the column, expr is the actual expression string with which we would like to filter, logic is 'AND' or 'OR', and cond is one of the following strings: "equals", "doesNotEqual", "contains", "doesNotContain", "greaterThan", "lessThan", "greaterThanOrEqualTo", "lessThanOrEqualTo", "true", "false", "null", "notNull", "empty", "notEmpty", "startsWith", "endsWith", "today", "yesterday", "on", "notOn", "thisMonth", "lastMonth", "nextMonth", "before", "after", "thisYear", "lastYear", "nextYear". The difference between the empty and null filtering conditions is that empty includes null, NaN, and undefined, as well as the empty string.
     * @param {?=} updateUI     specifies whether the filter row should be also updated once the grid is filtered
     * @param {?=} addedFromAdvanced
     * @return {?}
     */
    /* istanbul ignore next */
    IgGridFilteringFeature.prototype.filter = /**
     * Applies filtering programmatically and updates the UI by default.
     *
     * @param {?} expressions     An array of filtering expressions, each one having the format {fieldName: , expr: , cond: , logic: } where  fieldName is the key of the column, expr is the actual expression string with which we would like to filter, logic is 'AND' or 'OR', and cond is one of the following strings: "equals", "doesNotEqual", "contains", "doesNotContain", "greaterThan", "lessThan", "greaterThanOrEqualTo", "lessThanOrEqualTo", "true", "false", "null", "notNull", "empty", "notEmpty", "startsWith", "endsWith", "today", "yesterday", "on", "notOn", "thisMonth", "lastMonth", "nextMonth", "before", "after", "thisYear", "lastYear", "nextYear". The difference between the empty and null filtering conditions is that empty includes null, NaN, and undefined, as well as the empty string.
     * @param {?=} updateUI     specifies whether the filter row should be also updated once the grid is filtered
     * @param {?=} addedFromAdvanced
     * @return {?}
     */
    /* istanbul ignore next */
    function (expressions, updateUI, addedFromAdvanced) { return; };
    ;
    /**
     * Check whether filterCondition requires or not filtering expression - e.g. if filterCondition is "lastMonth", "thisMonth", "null", "notNull", "true", "false", etc. then filtering expression is NOT required
     *
     * @param filterCondition    filtering condition - e.g. "true", "false",  "yesterday", "empty", "null", etc.
     */
    /* istanbul ignore next */
    /**
     * Check whether filterCondition requires or not filtering expression - e.g. if filterCondition is "lastMonth", "thisMonth", "null", "notNull", "true", "false", etc. then filtering expression is NOT required
     *
     * @param {?} filterCondition    filtering condition - e.g. "true", "false",  "yesterday", "empty", "null", etc.
     * @return {?}
     */
    /* istanbul ignore next */
    IgGridFilteringFeature.prototype.requiresFilteringExpression = /**
     * Check whether filterCondition requires or not filtering expression - e.g. if filterCondition is "lastMonth", "thisMonth", "null", "notNull", "true", "false", etc. then filtering expression is NOT required
     *
     * @param {?} filterCondition    filtering condition - e.g. "true", "false",  "yesterday", "empty", "null", etc.
     * @return {?}
     */
    /* istanbul ignore next */
    function (filterCondition) { return; };
    ;
    IgGridFilteringFeature.decorators = [
        { type: Directive, args: [{
                    selector: 'filtering',
                    inputs: ["disabled", "create", "caseSensitive", "filterSummaryAlwaysVisible", "renderFC", "filterSummaryTemplate", "filterDropDownAnimations", "filterDropDownAnimationDuration", "filterDropDownWidth", "filterDropDownHeight", "filterExprUrlKey", "filterDropDownItemIcons", "columnSettings", "type", "filterDelay", "mode", "advancedModeEditorsVisible", "advancedModeHeaderButtonLocation", "filterDialogWidth", "filterDialogHeight", "filterDialogFilterDropDownDefaultWidth", "filterDialogExprInputDefaultWidth", "filterDialogColumnDropDownDefaultWidth", "renderFilterButton", "filterButtonLocation", "nullTexts", "labels", "tooltipTemplate", "filterDialogAddConditionTemplate", "filterDialogAddConditionDropDownTemplate", "filterDialogFilterTemplate", "filterDialogFilterConditionTemplate", "filterDialogAddButtonWidth", "filterDialogOkCancelButtonWidth", "filterDialogMaxFilterCount", "filterDialogContainment", "showEmptyConditions", "showNullConditions", "featureChooserText", "featureChooserTextHide", "featureChooserTextAdvancedFilter", "dialogWidget", "persist", "inherit"],
                    outputs: ["dataFiltering", "dataFiltered", "dropDownOpening", "dropDownOpened", "dropDownClosing", "dropDownClosed", "filterDialogOpening", "filterDialogOpened", "filterDialogMoving", "filterDialogFilterAdding", "filterDialogFilterAdded", "filterDialogClosing", "filterDialogClosed", "filterDialogContentsRendering", "filterDialogContentsRendered", "filterDialogFiltering"]
                },] }
    ];
    /** @nocollapse */
    IgGridFilteringFeature.ctorParameters = function () { return [
        { type: ElementRef }
    ]; };
    return IgGridFilteringFeature;
}(Feature));
export { IgGridFilteringFeature };
