// igniteui-angular-wrappers/esm5/lib/iggrid/iggridfeatures/iggridresizing.directive.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { ElementRef, Directive } from "@angular/core";
import { Feature } from "../feature";
var IgGridResizingFeature = /** @class */ (function (_super) {
    tslib_1.__extends(IgGridResizingFeature, _super);
    function IgGridResizingFeature(el) {
        return _super.call(this, el) || this;
    }
    /**
     * Destroys the resizing widget
     */
    /* istanbul ignore next */
    /**
     * Destroys the resizing widget
     * @return {?}
     */
    /* istanbul ignore next */
    IgGridResizingFeature.prototype.destroy = /**
     * Destroys the resizing widget
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    /**
     * Resizes a column to a specified width in pixels, percents or auto if no width is specified.
     *
     * @param column    An identifier for the column. If a number is provided it will be used as a columnIndex else if a strings is provided it will be used as a columnKey.
     * @param width    Width of the column in pixels or percents. If no width or "*" is specified the column will be auto-sized to the width of the data in it (including header and footer cells).
     */
    /* istanbul ignore next */
    /**
     * Resizes a column to a specified width in pixels, percents or auto if no width is specified.
     *
     * @param {?} column    An identifier for the column. If a number is provided it will be used as a columnIndex else if a strings is provided it will be used as a columnKey.
     * @param {?=} width    Width of the column in pixels or percents. If no width or "*" is specified the column will be auto-sized to the width of the data in it (including header and footer cells).
     * @return {?}
     */
    /* istanbul ignore next */
    IgGridResizingFeature.prototype.resize = /**
     * Resizes a column to a specified width in pixels, percents or auto if no width is specified.
     *
     * @param {?} column    An identifier for the column. If a number is provided it will be used as a columnIndex else if a strings is provided it will be used as a columnKey.
     * @param {?=} width    Width of the column in pixels or percents. If no width or "*" is specified the column will be auto-sized to the width of the data in it (including header and footer cells).
     * @return {?}
     */
    /* istanbul ignore next */
    function (column, width) { return; };
    ;
    IgGridResizingFeature.decorators = [
        { type: Directive, args: [{
                    selector: 'resizing',
                    inputs: ["disabled", "create", "allowDoubleClickToResize", "deferredResizing", "columnSettings", "handleThreshold", "inherit"],
                    outputs: ["columnResizing", "columnResizingRefused", "columnResized"]
                },] }
    ];
    /** @nocollapse */
    IgGridResizingFeature.ctorParameters = function () { return [
        { type: ElementRef }
    ]; };
    return IgGridResizingFeature;
}(Feature));
export { IgGridResizingFeature };
