// igniteui-angular-wrappers/esm5/lib/iggrid/iggridfeatures/iggridtooltips.directive.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Feature } from "../feature";
import { ElementRef, Directive } from "@angular/core";
var IgGridTooltipsFeature = /** @class */ (function (_super) {
    tslib_1.__extends(IgGridTooltipsFeature, _super);
    function IgGridTooltipsFeature(el) {
        return _super.call(this, el) || this;
    }
    /**
     * Destroys the tooltip widget.
     */
    /* istanbul ignore next */
    /**
     * Destroys the tooltip widget.
     * @return {?}
     */
    /* istanbul ignore next */
    IgGridTooltipsFeature.prototype.destroy = /**
     * Destroys the tooltip widget.
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    /**
     * Returns the ID of the parent div element bounding the ruler and the tooltip container
     */
    /* istanbul ignore next */
    /**
     * Returns the ID of the parent div element bounding the ruler and the tooltip container
     * @return {?}
     */
    /* istanbul ignore next */
    IgGridTooltipsFeature.prototype.id = /**
     * Returns the ID of the parent div element bounding the ruler and the tooltip container
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    IgGridTooltipsFeature.decorators = [
        { type: Directive, args: [{
                    selector: 'tooltips',
                    inputs: ["disabled", "create", "visibility", "style", "showDelay", "hideDelay", "columnSettings", "fadeTimespan", "cursorLeftOffset", "cursorTopOffset", "inherit"],
                    outputs: ["tooltipShowing", "tooltipShown", "tooltipHiding", "tooltipHidden"]
                },] }
    ];
    /** @nocollapse */
    IgGridTooltipsFeature.ctorParameters = function () { return [
        { type: ElementRef }
    ]; };
    return IgGridTooltipsFeature;
}(Feature));
export { IgGridTooltipsFeature };
