// igniteui-angular-wrappers/esm5/lib/iggrid/iggridfeatures/iggridmulticolumnheaders.directive.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { ElementRef, Directive } from "@angular/core";
import { Feature } from "../feature";
var IgGridMultiColumnHeadersFeature = /** @class */ (function (_super) {
    tslib_1.__extends(IgGridMultiColumnHeadersFeature, _super);
    function IgGridMultiColumnHeadersFeature(el) {
        return _super.call(this, el) || this;
    }
    /**
     * Destroys the widget
     */
    /* istanbul ignore next */
    /**
     * Destroys the widget
     * @return {?}
     */
    /* istanbul ignore next */
    IgGridMultiColumnHeadersFeature.prototype.destroy = /**
     * Destroys the widget
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    /**
     * Collapses an expanded group. If the group is collapsed, the method does nothing.
     * Note: This method is asynchronous which means that it returns immediately and any subsequent code will execute in parallel. This may lead to runtime errors. To avoid them put the subsequent code in the callback parameter provided by the method.
     */
    /* istanbul ignore next */
    /**
     * Collapses an expanded group. If the group is collapsed, the method does nothing.
     * Note: This method is asynchronous which means that it returns immediately and any subsequent code will execute in parallel. This may lead to runtime errors. To avoid them put the subsequent code in the callback parameter provided by the method.
     * @param {?} groupKey
     * @param {?=} callback
     * @return {?}
     */
    /* istanbul ignore next */
    IgGridMultiColumnHeadersFeature.prototype.collapseGroup = /**
     * Collapses an expanded group. If the group is collapsed, the method does nothing.
     * Note: This method is asynchronous which means that it returns immediately and any subsequent code will execute in parallel. This may lead to runtime errors. To avoid them put the subsequent code in the callback parameter provided by the method.
     * @param {?} groupKey
     * @param {?=} callback
     * @return {?}
     */
    /* istanbul ignore next */
    function (groupKey, callback) { return; };
    ;
    /**
     * Expands a collapsed group. If the group is expanded, the method does nothing.
     * Note: This method is asynchronous which means that it returns immediately and any subsequent code will execute in parallel. This may lead to runtime errors. To avoid them put the subsequent code in the callback parameter provided by the method.
     */
    /* istanbul ignore next */
    /**
     * Expands a collapsed group. If the group is expanded, the method does nothing.
     * Note: This method is asynchronous which means that it returns immediately and any subsequent code will execute in parallel. This may lead to runtime errors. To avoid them put the subsequent code in the callback parameter provided by the method.
     * @param {?} groupKey
     * @param {?=} callback
     * @return {?}
     */
    /* istanbul ignore next */
    IgGridMultiColumnHeadersFeature.prototype.expandGroup = /**
     * Expands a collapsed group. If the group is expanded, the method does nothing.
     * Note: This method is asynchronous which means that it returns immediately and any subsequent code will execute in parallel. This may lead to runtime errors. To avoid them put the subsequent code in the callback parameter provided by the method.
     * @param {?} groupKey
     * @param {?=} callback
     * @return {?}
     */
    /* istanbul ignore next */
    function (groupKey, callback) { return; };
    ;
    /**
     * Returns multicolumn headers array. if there aren"t multicolumn headers returns undefined.
     */
    /* istanbul ignore next */
    /**
     * Returns multicolumn headers array. if there aren"t multicolumn headers returns undefined.
     * @return {?}
     */
    /* istanbul ignore next */
    IgGridMultiColumnHeadersFeature.prototype.getMultiColumnHeaders = /**
     * Returns multicolumn headers array. if there aren"t multicolumn headers returns undefined.
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    /**
     * Toggles a collapsible group.
     * Note: This method is asynchronous which means that it returns immediately and any subsequent code will execute in parallel. This may lead to runtime errors. To avoid them put the subsequent code in the callback parameter provided by the method.
     */
    /* istanbul ignore next */
    /**
     * Toggles a collapsible group.
     * Note: This method is asynchronous which means that it returns immediately and any subsequent code will execute in parallel. This may lead to runtime errors. To avoid them put the subsequent code in the callback parameter provided by the method.
     * @param {?} groupKey
     * @param {?=} callback
     * @return {?}
     */
    /* istanbul ignore next */
    IgGridMultiColumnHeadersFeature.prototype.toggleGroup = /**
     * Toggles a collapsible group.
     * Note: This method is asynchronous which means that it returns immediately and any subsequent code will execute in parallel. This may lead to runtime errors. To avoid them put the subsequent code in the callback parameter provided by the method.
     * @param {?} groupKey
     * @param {?=} callback
     * @return {?}
     */
    /* istanbul ignore next */
    function (groupKey, callback) { return; };
    ;
    IgGridMultiColumnHeadersFeature.decorators = [
        { type: Directive, args: [{
                    selector: 'multi-column-headers',
                    inputs: ["inherit"],
                    outputs: ["groupCollapsing", "groupCollapsed", "groupExpanding", "groupExpanded"]
                },] }
    ];
    /** @nocollapse */
    IgGridMultiColumnHeadersFeature.ctorParameters = function () { return [
        { type: ElementRef }
    ]; };
    return IgGridMultiColumnHeadersFeature;
}(Feature));
export { IgGridMultiColumnHeadersFeature };
