// igniteui-angular-wrappers/esm5/lib/iggrid/iggridfeatures/iggridappendrowsondemand.directive.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Feature } from "../feature";
import { ElementRef, Directive } from "@angular/core";
var IgGridAppendRowsOnDemandFeature = /** @class */ (function (_super) {
    tslib_1.__extends(IgGridAppendRowsOnDemandFeature, _super);
    function IgGridAppendRowsOnDemandFeature(el) {
        return _super.call(this, el) || this;
    }
    /**
     * Destroys the append rows on demand widget
     */
    /* istanbul ignore next */
    /**
     * Destroys the append rows on demand widget
     * @return {?}
     */
    /* istanbul ignore next */
    IgGridAppendRowsOnDemandFeature.prototype.destroy = /**
     * Destroys the append rows on demand widget
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    /**
     * Loads the next chunk of data.
     */
    /* istanbul ignore next */
    /**
     * Loads the next chunk of data.
     * @return {?}
     */
    /* istanbul ignore next */
    IgGridAppendRowsOnDemandFeature.prototype.nextChunk = /**
     * Loads the next chunk of data.
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    IgGridAppendRowsOnDemandFeature.decorators = [
        { type: Directive, args: [{
                    selector: 'append-rows-on-demand',
                    inputs: ["type", "chunkSize", "recordCountKey", "chunkSizeUrlKey", "chunkIndexUrlKey", "defaultChunkIndex", "currentChunkIndex", "loadTrigger", "loadMoreDataButtonText"],
                    outputs: ["rowsRequesting", "rowsRequested"]
                },] }
    ];
    /** @nocollapse */
    IgGridAppendRowsOnDemandFeature.ctorParameters = function () { return [
        { type: ElementRef }
    ]; };
    return IgGridAppendRowsOnDemandFeature;
}(Feature));
export { IgGridAppendRowsOnDemandFeature };
