// igniteui-angular-wrappers/esm5/lib/iggrid/iggridfeatures/iggridresponsive.directive.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Feature } from "../feature";
import { ElementRef, Directive } from "@angular/core";
var IgGridResponsiveFeature = /** @class */ (function (_super) {
    tslib_1.__extends(IgGridResponsiveFeature, _super);
    function IgGridResponsiveFeature(el) {
        return _super.call(this, el) || this;
    }
    /**
     * Destroys the responsive widget.
     */
    /* istanbul ignore next */
    /**
     * Destroys the responsive widget.
     * @return {?}
     */
    /* istanbul ignore next */
    IgGridResponsiveFeature.prototype.destroy = /**
     * Destroys the responsive widget.
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    /**
     * Returns the currently active responsive mode.
     */
    /* istanbul ignore next */
    /**
     * Returns the currently active responsive mode.
     * @return {?}
     */
    /* istanbul ignore next */
    IgGridResponsiveFeature.prototype.getCurrentResponsiveMode = /**
     * Returns the currently active responsive mode.
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    IgGridResponsiveFeature.decorators = [
        { type: Directive, args: [{
                    selector: 'responsive',
                    inputs: ["disabled", "create", "columnSettings", "reactOnContainerWidthChanges", "forceResponsiveGridWidth", "responsiveSensitivity", "responsiveModes", "enableVerticalRendering", "windowWidthToRenderVertically", "propertiesColumnWidth", "valuesColumnWidth", "allowedColumnWidthPerType", "singleColumnTemplate", "inherit"],
                    outputs: ["responsiveColumnHiding", "responsiveColumnHidden", "responsiveColumnShowing", "responsiveColumnShown", "responsiveModeChanged"]
                },] }
    ];
    /** @nocollapse */
    IgGridResponsiveFeature.ctorParameters = function () { return [
        { type: ElementRef }
    ]; };
    return IgGridResponsiveFeature;
}(Feature));
export { IgGridResponsiveFeature };
