// igniteui-angular-wrappers/esm5/lib/igsplitbutton/igsplitbutton.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { ElementRef, Renderer, IterableDiffers, Component, KeyValueDiffers, ChangeDetectorRef } from "@angular/core";
import { IgControlBase } from "../igcontrolbase/igcontrolbase";
var IgSplitButtonComponent = /** @class */ (function (_super) {
    tslib_1.__extends(IgSplitButtonComponent, _super);
    function IgSplitButtonComponent(el, renderer, differs, kvalDiffers, cdr) {
        return _super.call(this, el, renderer, differs, kvalDiffers, cdr) || this;
    }
    /**
     * Switch to given igToolbar button.
     *
     * @param button . The button jQuery element.
     */
    /* istanbul ignore next */
    /**
     * Switch to given igToolbar button.
     *
     * @param {?} button . The button jQuery element.
     * @return {?}
     */
    /* istanbul ignore next */
    IgSplitButtonComponent.prototype.switchToButton = /**
     * Switch to given igToolbar button.
     *
     * @param {?} button . The button jQuery element.
     * @return {?}
     */
    /* istanbul ignore next */
    function (button) { return; };
    ;
    /**
     * Toggle widget state between collapsed and expanded.
     *
     * @param e The igSplitButton to be expanded/collapsed.
     */
    /* istanbul ignore next */
    /**
     * Toggle widget state between collapsed and expanded.
     *
     * @param {?} e The igSplitButton to be expanded/collapsed.
     * @return {?}
     */
    /* istanbul ignore next */
    IgSplitButtonComponent.prototype.toggle = /**
     * Toggle widget state between collapsed and expanded.
     *
     * @param {?} e The igSplitButton to be expanded/collapsed.
     * @return {?}
     */
    /* istanbul ignore next */
    function (e) { return; };
    ;
    /**
     * Collapse the widget.
     *
     * @param e Indicates the browser event which triggered this action (not API).
     */
    /* istanbul ignore next */
    /**
     * Collapse the widget.
     *
     * @param {?=} e Indicates the browser event which triggered this action (not API).
     * @return {?}
     */
    /* istanbul ignore next */
    IgSplitButtonComponent.prototype.collapse = /**
     * Collapse the widget.
     *
     * @param {?=} e Indicates the browser event which triggered this action (not API).
     * @return {?}
     */
    /* istanbul ignore next */
    function (e) { return; };
    ;
    /**
     * Expands the widget.
     *
     * @param e Indicates the browser event which triggered this action (not API).
     */
    /* istanbul ignore next */
    /**
     * Expands the widget.
     *
     * @param {?=} e Indicates the browser event which triggered this action (not API).
     * @return {?}
     */
    /* istanbul ignore next */
    IgSplitButtonComponent.prototype.expand = /**
     * Expands the widget.
     *
     * @param {?=} e Indicates the browser event which triggered this action (not API).
     * @return {?}
     */
    /* istanbul ignore next */
    function (e) { return; };
    ;
    /**
     * Destroy the widget.
     */
    /* istanbul ignore next */
    /**
     * Destroy the widget.
     * @return {?}
     */
    /* istanbul ignore next */
    IgSplitButtonComponent.prototype.destroy = /**
     * Destroy the widget.
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    /* istanbul ignore next */
    /* istanbul ignore next */
    /**
     * @return {?}
     */
    IgSplitButtonComponent.prototype.widget = /* istanbul ignore next */
    /**
     * @return {?}
     */
    function () { return; };
    ;
    IgSplitButtonComponent.decorators = [
        { type: Component, args: [{
                    selector: "ig-split-button",
                    template: "<ng-content></ng-content>",
                    inputs: ["widgetId", "options", "changeDetectionInterval", "disabled", "create", "items", "defaultItemName", "swapDefaultEnabled"],
                    outputs: ["click", "expanded", "expanding", "collapsed", "collapsing"]
                }] }
    ];
    /** @nocollapse */
    IgSplitButtonComponent.ctorParameters = function () { return [
        { type: ElementRef },
        { type: Renderer },
        { type: IterableDiffers },
        { type: KeyValueDiffers },
        { type: ChangeDetectorRef }
    ]; };
    return IgSplitButtonComponent;
}(IgControlBase));
export { IgSplitButtonComponent };
