// igniteui-angular-wrappers/esm5/lib/igrating/igrating.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Component, ElementRef, Renderer, IterableDiffers, KeyValueDiffers, ChangeDetectorRef } from "@angular/core";
import { IgControlBase } from "../igcontrolbase/igcontrolbase";
var IgRatingComponent = /** @class */ (function (_super) {
    tslib_1.__extends(IgRatingComponent, _super);
    function IgRatingComponent(el, renderer, differs, kvalDiffers, cdr) {
        return _super.call(this, el, renderer, differs, kvalDiffers, cdr) || this;
    }
    /**
     * Gets reference to [igValidator](ui.igvalidator) used by igRating.
     *
     * @param destroy     Request to destroy validator.
     */
    /* istanbul ignore next */
    /**
     * Gets reference to [igValidator](ui.igvalidator) used by igRating.
     *
     * @param {?=} destroy     Request to destroy validator.
     * @return {?}
     */
    /* istanbul ignore next */
    IgRatingComponent.prototype.validator = /**
     * Gets reference to [igValidator](ui.igvalidator) used by igRating.
     *
     * @param {?=} destroy     Request to destroy validator.
     * @return {?}
     */
    /* istanbul ignore next */
    function (destroy) { return; };
    ;
    /**
     * Triggers validation.
     */
    /* istanbul ignore next */
    /**
     * Triggers validation.
     * @return {?}
     */
    /* istanbul ignore next */
    IgRatingComponent.prototype.validate = /**
     * Triggers validation.
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    /**
     * Gets/Sets (selected) value.
     *
     * @param val     New value which is rendered with selected css.
     * @return number|object     If parameter is not 'number', then exact value rendered with selected css is returned. Otherwise, reference to igRating is returned.
     */
    /* istanbul ignore next */
    /**
     * Gets/Sets (selected) value.
     *
     * @param {?} val     New value which is rendered with selected css.
     * @return {?} number|object     If parameter is not 'number', then exact value rendered with selected css is returned. Otherwise, reference to igRating is returned.
     */
    /* istanbul ignore next */
    IgRatingComponent.prototype.value = /**
     * Gets/Sets (selected) value.
     *
     * @param {?} val     New value which is rendered with selected css.
     * @return {?} number|object     If parameter is not 'number', then exact value rendered with selected css is returned. Otherwise, reference to igRating is returned.
     */
    /* istanbul ignore next */
    function (val) { return; };
    ;
    /**
     * Gets/Sets hover value.
     *
     * @param val     New value which will be rendered with hover css when rating gets mouse.
     * @return number|object     If parameter is not "number", then last value which was rendered with hover css is returned. Otherwise, reference to igRating is returned.
     */
    /* istanbul ignore next */
    /**
     * Gets/Sets hover value.
     *
     * @param {?=} val     New value which will be rendered with hover css when rating gets mouse.
     * @return {?} number|object     If parameter is not "number", then last value which was rendered with hover css is returned. Otherwise, reference to igRating is returned.
     */
    /* istanbul ignore next */
    IgRatingComponent.prototype.valueHover = /**
     * Gets/Sets hover value.
     *
     * @param {?=} val     New value which will be rendered with hover css when rating gets mouse.
     * @return {?} number|object     If parameter is not "number", then last value which was rendered with hover css is returned. Otherwise, reference to igRating is returned.
     */
    /* istanbul ignore next */
    function (val) { return; };
    ;
    /**
     * Checks if igRating has focus.
     */
    /* istanbul ignore next */
    /**
     * Checks if igRating has focus.
     * @return {?}
     */
    /* istanbul ignore next */
    IgRatingComponent.prototype.hasFocus = /**
     * Checks if igRating has focus.
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    /**
     * Sets focus to igRating. That has effect only when options.focusable is enabled.
     */
    /* istanbul ignore next */
    /**
     * Sets focus to igRating. That has effect only when options.focusable is enabled.
     * @return {?}
     */
    /* istanbul ignore next */
    IgRatingComponent.prototype.focus = /**
     * Sets focus to igRating. That has effect only when options.focusable is enabled.
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    /**
     * Destroys igRating widget.
     */
    /* istanbul ignore next */
    /**
     * Destroys igRating widget.
     * @return {?}
     */
    /* istanbul ignore next */
    IgRatingComponent.prototype.destroy = /**
     * Destroys igRating widget.
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    IgRatingComponent.decorators = [
        { type: Component, args: [{
                    selector: "ig-rating",
                    template: "<ng-content></ng-content>",
                    inputs: ["widgetId", "options", "changeDetectionInterval", "disabled", "create", "vertical", "value", "valueHover", "voteCount", "voteWidth", "voteHeight", "swapDirection", "valueAsPercent", "focusable", "precision", "precisionZeroVote", "roundedDecimalPlaces", "theme", "validatorOptions", "cssVotes"],
                    outputs: ["hoverChange", "valueChange"]
                }] }
    ];
    /** @nocollapse */
    IgRatingComponent.ctorParameters = function () { return [
        { type: ElementRef },
        { type: Renderer },
        { type: IterableDiffers },
        { type: KeyValueDiffers },
        { type: ChangeDetectorRef }
    ]; };
    return IgRatingComponent;
}(IgControlBase));
export { IgRatingComponent };
