// igniteui-angular-wrappers/esm5/lib/igzoombar/igzoombar.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Component, ElementRef, Renderer, IterableDiffers, KeyValueDiffers, ChangeDetectorRef } from "@angular/core";
import { IgControlBase } from "../igcontrolbase/igcontrolbase";
var IgZoombarComponent = /** @class */ (function (_super) {
    tslib_1.__extends(IgZoombarComponent, _super);
    function IgZoombarComponent(el, renderer, differs, kvalDiffers, cdr) {
        return _super.call(this, el, renderer, differs, kvalDiffers, cdr) || this;
    }
    /**
     * Destroys the Zoombar widget
     */
    /* istanbul ignore next */
    /**
     * Destroys the Zoombar widget
     * @return {?}
     */
    /* istanbul ignore next */
    IgZoombarComponent.prototype.destroy = /**
     * Destroys the Zoombar widget
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    /**
     * Returns the main zoombar element
     */
    /* istanbul ignore next */
    /**
     * Returns the main zoombar element
     * @return {?}
     */
    /* istanbul ignore next */
    IgZoombarComponent.prototype.widget = /**
     * Returns the main zoombar element
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    /**
     * Returns the ID of the element the zoombar is initialized on
     */
    /* istanbul ignore next */
    /**
     * Returns the ID of the element the zoombar is initialized on
     * @return {?}
     */
    /* istanbul ignore next */
    IgZoombarComponent.prototype.id = /**
     * Returns the ID of the element the zoombar is initialized on
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    /**
     * Returns the DIV that is the topmost container of the zoombar widget
     */
    /* istanbul ignore next */
    /**
     * Returns the DIV that is the topmost container of the zoombar widget
     * @return {?}
     */
    /* istanbul ignore next */
    IgZoombarComponent.prototype.container = /**
     * Returns the DIV that is the topmost container of the zoombar widget
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    /**
     * Returns the element the clone widget is initialized on
     */
    /* istanbul ignore next */
    /**
     * Returns the element the clone widget is initialized on
     * @return {?}
     */
    /* istanbul ignore next */
    IgZoombarComponent.prototype.clone = /**
     * Returns the element the clone widget is initialized on
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    /**
     * Gets or sets the current zoom window
     *
     * @param left     The left parameter of the new zoom window in percentages
     * @param width     The width parameter of the new zoom window in percentages
     */
    /* istanbul ignore next */
    /**
     * Gets or sets the current zoom window
     *
     * @param {?=} left     The left parameter of the new zoom window in percentages
     * @param {?=} width     The width parameter of the new zoom window in percentages
     * @return {?}
     */
    /* istanbul ignore next */
    IgZoombarComponent.prototype.zoom = /**
     * Gets or sets the current zoom window
     *
     * @param {?=} left     The left parameter of the new zoom window in percentages
     * @param {?=} width     The width parameter of the new zoom window in percentages
     * @return {?}
     */
    /* istanbul ignore next */
    function (left, width) { return; };
    ;
    IgZoombarComponent.decorators = [
        { type: Component, args: [{
                    selector: "ig-zoombar",
                    template: "<ng-content></ng-content>",
                    inputs: ["widgetId", "options", "changeDetectionInterval", "disabled", "create", "target", "clone", "width", "height", "zoomAction", "zoomWindowMoveDistance", "defaultZoomWindow", "zoomWindowMinWidth", "hoverStyleAnimationDuration", "windowPanDuration", "tabIndex"],
                    outputs: ["zoomChanging", "zoomChanged", "providerCreated", "windowDragStarting", "windowDragStarted", "windowDragging", "windowDragEnding", "windowDragEnded", "windowResizing", "windowResized"]
                }] }
    ];
    /** @nocollapse */
    IgZoombarComponent.ctorParameters = function () { return [
        { type: ElementRef },
        { type: Renderer },
        { type: IterableDiffers },
        { type: KeyValueDiffers },
        { type: ChangeDetectorRef }
    ]; };
    return IgZoombarComponent;
}(IgControlBase));
export { IgZoombarComponent };
