// igniteui-angular-wrappers/esm5/lib/igdoughnutchart/igdoughnutchart.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Component, ElementRef, Renderer, IterableDiffers, KeyValueDiffers, ChangeDetectorRef, Input } from "@angular/core";
import { IgControlBase } from "../igcontrolbase/igcontrolbase";
var IgDoughnutChartComponent = /** @class */ (function (_super) {
    tslib_1.__extends(IgDoughnutChartComponent, _super);
    function IgDoughnutChartComponent(el, renderer, differs, kvalDiffers, cdr) {
        return _super.call(this, el, renderer, differs, kvalDiffers, cdr) || this;
    }
    Object.defineProperty(IgDoughnutChartComponent.prototype, "dataSource", {
        set: /**
         * @param {?} value
         * @return {?}
         */
        function (value) {
            this._dataSource = value;
            /** @type {?} */
            var chart = jQuery(this._el).data(this._widgetName);
            if (chart) {
                jQuery(this._el)[this._widgetName]("option", "dataSource", this._dataSource);
            }
        },
        enumerable: true,
        configurable: true
    });
    ;
    /**
     * @return {?}
     */
    IgDoughnutChartComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        if (this._dataSource === null || this._dataSource === undefined) {
            this._dataSource = this.options["dataSource"];
        }
        if (!this.options["dataSource"] && this._dataSource) {
            this.options["dataSource"] = this._dataSource;
        }
        _super.prototype.ngOnInit.call(this);
    };
    /**
     * Adds a new series to the doughnut chart.
     *
     * @param seriesObj     The series object to be added.
     */
    /* istanbul ignore next */
    /**
     * Adds a new series to the doughnut chart.
     *
     * @param {?} seriesObj     The series object to be added.
     * @return {?}
     */
    /* istanbul ignore next */
    IgDoughnutChartComponent.prototype.addSeries = /**
     * Adds a new series to the doughnut chart.
     *
     * @param {?} seriesObj     The series object to be added.
     * @return {?}
     */
    /* istanbul ignore next */
    function (seriesObj) { return; };
    ;
    /**
     * Removes the specified series from the doughnut chart.
     *
     * @param seriesObj     The series object identifying the series to be removed.
     */
    /* istanbul ignore next */
    /**
     * Removes the specified series from the doughnut chart.
     *
     * @param {?} seriesObj     The series object identifying the series to be removed.
     * @return {?}
     */
    /* istanbul ignore next */
    IgDoughnutChartComponent.prototype.removeSeries = /**
     * Removes the specified series from the doughnut chart.
     *
     * @param {?} seriesObj     The series object identifying the series to be removed.
     * @return {?}
     */
    /* istanbul ignore next */
    function (seriesObj) { return; };
    ;
    /**
     * Updates the series with the specified name with the specified new property values.
     *
     * @param value     The series object identifying the series to be updated.
     */
    /* istanbul ignore next */
    /**
     * Updates the series with the specified name with the specified new property values.
     *
     * @param {?} value     The series object identifying the series to be updated.
     * @return {?}
     */
    /* istanbul ignore next */
    IgDoughnutChartComponent.prototype.updateSeries = /**
     * Updates the series with the specified name with the specified new property values.
     *
     * @param {?} value     The series object identifying the series to be updated.
     * @return {?}
     */
    /* istanbul ignore next */
    function (value) { return; };
    ;
    /**
     * Returns the center of the doughnut chart.
     */
    /* istanbul ignore next */
    /**
     * Returns the center of the doughnut chart.
     * @return {?}
     */
    /* istanbul ignore next */
    IgDoughnutChartComponent.prototype.getCenterCoordinates = /**
     * Returns the center of the doughnut chart.
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    /**
     * Returns the radius of the chart's hole.
     */
    /* istanbul ignore next */
    /**
     * Returns the radius of the chart's hole.
     * @return {?}
     */
    /* istanbul ignore next */
    IgDoughnutChartComponent.prototype.getHoleRadius = /**
     * Returns the radius of the chart's hole.
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    /**
     * Returns information about how the doughnut chart is rendered.
     */
    /* istanbul ignore next */
    /**
     * Returns information about how the doughnut chart is rendered.
     * @return {?}
     */
    /* istanbul ignore next */
    IgDoughnutChartComponent.prototype.exportVisualData = /**
     * Returns information about how the doughnut chart is rendered.
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    /**
     * Causes all of the series that have pending changes e.g. by changed property values to be rendered immediately.
     */
    /* istanbul ignore next */
    /**
     * Causes all of the series that have pending changes e.g. by changed property values to be rendered immediately.
     * @return {?}
     */
    /* istanbul ignore next */
    IgDoughnutChartComponent.prototype.flush = /**
     * Causes all of the series that have pending changes e.g. by changed property values to be rendered immediately.
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    /**
     * Destroys the widget.
     */
    /* istanbul ignore next */
    /**
     * Destroys the widget.
     * @return {?}
     */
    /* istanbul ignore next */
    IgDoughnutChartComponent.prototype.destroy = /**
     * Destroys the widget.
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    IgDoughnutChartComponent.decorators = [
        { type: Component, args: [{
                    selector: "ig-doughnut-chart",
                    template: "<ng-content></ng-content>",
                    inputs: ["widgetId", "options", "changeDetectionInterval", "disabled", "create", "width", "height", "tooltipTemplate", "maxRecCount", "dataSource", "dataSourceType", "dataSourceUrl", "responseTotalRecCountKey", "responseDataKey", "series", "allowSliceSelection", "isSurfaceInteractionDisabled", "allowSliceExplosion", "innerExtent", "selectedStyle"],
                    outputs: ["dataBinding", "dataBound", "updateTooltip", "hideTooltip", "tooltipShowing", "tooltipShown", "tooltipHiding", "tooltipHidden", "browserNotSupported", "sliceClick", "holeDimensionsChanged"]
                }] }
    ];
    /** @nocollapse */
    IgDoughnutChartComponent.ctorParameters = function () { return [
        { type: ElementRef },
        { type: Renderer },
        { type: IterableDiffers },
        { type: KeyValueDiffers },
        { type: ChangeDetectorRef }
    ]; };
    IgDoughnutChartComponent.propDecorators = {
        dataSource: [{ type: Input }]
    };
    return IgDoughnutChartComponent;
}(IgControlBase));
export { IgDoughnutChartComponent };
if (false) {
    /**
     * @type {?}
     * @private
     */
    IgDoughnutChartComponent.prototype._dataSource;
    /* Skipping unhandled member: ;*/
    /* Skipping unhandled member: ;*/
    /* Skipping unhandled member: ;*/
    /* Skipping unhandled member: ;*/
    /* Skipping unhandled member: ;*/
    /* Skipping unhandled member: ;*/
    /* Skipping unhandled member: ;*/
    /* Skipping unhandled member: ;*/
    /* Skipping unhandled member: ;*/
}
