// igniteui-angular-wrappers/esm5/lib/igspreadsheet/igspreadsheet.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Component, ElementRef, Renderer, IterableDiffers, KeyValueDiffers, ChangeDetectorRef } from "@angular/core";
import { IgControlBase } from "../igcontrolbase/igcontrolbase";
var IgSpreadsheetComponent = /** @class */ (function (_super) {
    tslib_1.__extends(IgSpreadsheetComponent, _super);
    function IgSpreadsheetComponent(el, renderer, differs, kvalDiffers, cdr) {
        return _super.call(this, el, renderer, differs, kvalDiffers, cdr) || this;
    }
    /**
     * Returns an object that represents the pane with the focus.
     */
    /* istanbul ignore next */
    /**
     * Returns an object that represents the pane with the focus.
     * @return {?}
     */
    /* istanbul ignore next */
    IgSpreadsheetComponent.prototype.getActivePane = /**
     * Returns an object that represents the pane with the focus.
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    /**
     * Returns an object that represents the current selection of the active pane.
     */
    /* istanbul ignore next */
    /**
     * Returns an object that represents the current selection of the active pane.
     * @return {?}
     */
    /* istanbul ignore next */
    IgSpreadsheetComponent.prototype.getActiveSelection = /**
     * Returns an object that represents the current selection of the active pane.
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    /**
     * Returns an object used to get the formatting of the activeCell and where modifications are applied to the entire active selection.
     *				Any changes made to this object will affect all the objects in the selection. So for example, the
    *				Font.Name may return "Arial" because the active cell has that as its resolved font name even though the other
    *				cells are using a different font but if you set the Font.Name of this object to "Arial" then all the objects
    *				affected by the selection will have their Font.Name updated to that value.
    */
    /* istanbul ignore next */
    /**
     * Returns an object used to get the formatting of the activeCell and where modifications are applied to the entire active selection.
     * 				Any changes made to this object will affect all the objects in the selection. So for example, the
     * 				Font.Name may return "Arial" because the active cell has that as its resolved font name even though the other
     * 				cells are using a different font but if you set the Font.Name of this object to "Arial" then all the objects
     * 				affected by the selection will have their Font.Name updated to that value.
     * @return {?}
     */
    /* istanbul ignore next */
    IgSpreadsheetComponent.prototype.getActiveSelectionCellRangeFormat = /**
     * Returns an object used to get the formatting of the activeCell and where modifications are applied to the entire active selection.
     * 				Any changes made to this object will affect all the objects in the selection. So for example, the
     * 				Font.Name may return "Arial" because the active cell has that as its resolved font name even though the other
     * 				cells are using a different font but if you set the Font.Name of this object to "Arial" then all the objects
     * 				affected by the selection will have their Font.Name updated to that value.
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    /**
     * Returns a boolean indicating if the user is currently editing the name of the active worksheet.
     */
    /* istanbul ignore next */
    /**
     * Returns a boolean indicating if the user is currently editing the name of the active worksheet.
     * @return {?}
     */
    /* istanbul ignore next */
    IgSpreadsheetComponent.prototype.getIsRenamingWorksheet = /**
     * Returns a boolean indicating if the user is currently editing the name of the active worksheet.
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    /**
     * Returns an array of the panes for the activeWorksheet.
     *
    *				 returnType="ig.spreadsheet.SpreadsheetPane[]"
    */
    /* istanbul ignore next */
    /**
     * Returns an array of the panes for the activeWorksheet.
     *
     * 				 returnType="ig.spreadsheet.SpreadsheetPane[]"
     * @return {?}
     */
    /* istanbul ignore next */
    IgSpreadsheetComponent.prototype.getPanes = /**
     * Returns an array of the panes for the activeWorksheet.
     *
     * 				 returnType="ig.spreadsheet.SpreadsheetPane[]"
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    /**
     * Executes the action associated with the specified id.
     *
     * @param action An [enumeration](ig.spreadsheet.SpreadsheetAction) or string that identifies the action to execute.
     */
    /* istanbul ignore next */
    /**
     * Executes the action associated with the specified id.
     *
     * @param {?} action An [enumeration](ig.spreadsheet.SpreadsheetAction) or string that identifies the action to execute.
     * @return {?}
     */
    /* istanbul ignore next */
    IgSpreadsheetComponent.prototype.executeAction = /**
     * Executes the action associated with the specified id.
     *
     * @param {?} action An [enumeration](ig.spreadsheet.SpreadsheetAction) or string that identifies the action to execute.
     * @return {?}
     */
    /* istanbul ignore next */
    function (action) { return; };
    ;
    /**
     * Forces any pending deferred work to render on the spreadsheet before continuing
     */
    /* istanbul ignore next */
    /**
     * Forces any pending deferred work to render on the spreadsheet before continuing
     * @return {?}
     */
    /* istanbul ignore next */
    IgSpreadsheetComponent.prototype.flush = /**
     * Forces any pending deferred work to render on the spreadsheet before continuing
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    /**
     * Destroys the widget.
     */
    /* istanbul ignore next */
    /**
     * Destroys the widget.
     * @return {?}
     */
    /* istanbul ignore next */
    IgSpreadsheetComponent.prototype.destroy = /**
     * Destroys the widget.
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    /**
     * Notify the spreadsheet that style information used for rendering the spreadsheet may have been updated.
     */
    /* istanbul ignore next */
    /**
     * Notify the spreadsheet that style information used for rendering the spreadsheet may have been updated.
     * @return {?}
     */
    /* istanbul ignore next */
    IgSpreadsheetComponent.prototype.styleUpdated = /**
     * Notify the spreadsheet that style information used for rendering the spreadsheet may have been updated.
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    IgSpreadsheetComponent.decorators = [
        { type: Component, args: [{
                    selector: "ig-spreadsheet",
                    template: "<ng-content></ng-content>",
                    inputs: ["widgetId", "options", "changeDetectionInterval", "disabled", "create", "width", "height", "activeCell", "isScrollLocked", "activeWorksheet", "allowAddWorksheet", "allowDeleteWorksheet", "areGridlinesVisible", "areHeadersVisible", "enterKeyNavigationDirection", "isEnterKeyNavigationEnabled", "isFormulaBarVisible", "isInEndMode", "isUndoEnabled", "nameBoxWidth", "selectionMode", "selectedWorksheets", "validationInputMessagePosition", "workbook", "zoomLevel"],
                    outputs: ["actionExecuted", "actionExecuting", "activeCellChanged", "activePaneChanged", "activeWorksheetChanged", "editRangePasswordNeeded", "hyperlinkExecuting", "selectionChanged", "userPromptDisplaying", "workbookDirtied"]
                }] }
    ];
    /** @nocollapse */
    IgSpreadsheetComponent.ctorParameters = function () { return [
        { type: ElementRef },
        { type: Renderer },
        { type: IterableDiffers },
        { type: KeyValueDiffers },
        { type: ChangeDetectorRef }
    ]; };
    return IgSpreadsheetComponent;
}(IgControlBase));
export { IgSpreadsheetComponent };
