// igniteui-angular-wrappers/esm5/lib/igupload/igupload.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Component, ElementRef, Renderer, IterableDiffers, KeyValueDiffers, ChangeDetectorRef } from "@angular/core";
import { IgControlBase } from "../igcontrolbase/igcontrolbase";
var IgUploadComponent = /** @class */ (function (_super) {
    tslib_1.__extends(IgUploadComponent, _super);
    function IgUploadComponent(el, renderer, differs, kvalDiffers, cdr) {
        return _super.call(this, el, renderer, differs, kvalDiffers, cdr) || this;
    }
    /**
     * Return jquery object of fileupload container - html DOM element
     */
    /* istanbul ignore next */
    /**
     * Return jquery object of fileupload container - html DOM element
     * @return {?}
     */
    /* istanbul ignore next */
    IgUploadComponent.prototype.container = /**
     * Return jquery object of fileupload container - html DOM element
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    /**
     * Returns the current widget element
     */
    /* istanbul ignore next */
    /**
     * Returns the current widget element
     * @return {?}
     */
    /* istanbul ignore next */
    IgUploadComponent.prototype.widget = /**
     * Returns the current widget element
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    /**
     * Hide finished files
     */
    /* istanbul ignore next */
    /**
     * Hide finished files
     * @return {?}
     */
    /* istanbul ignore next */
    IgUploadComponent.prototype.clearAll = /**
     * Hide finished files
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    /**
     * Append additional data field to formData(before submitting it to the server). Usually this function is used in the handler of the event onFormDataSubmit. If the browser supports HTML5 file API formData is instance of FormData, otherwise(like IE10 and older) formData is jQuery representation of the <form> that should be submitted to the server
     *
     * @param formData    If the browser supports HTML5 file API formData is instance of FormData, otherwise(like IE10 and older) formData is jQuery representation of the <form> that should be submitted to the server
     * @param field    Data field that should be appended to the formData. The object has 2 properties - value and name. If the browser supports HTML5 the data field is appended to the formData object. Otherwise it is appended as input hidden field to the <form>
     */
    /* istanbul ignore next */
    /**
     * Append additional data field to formData(before submitting it to the server). Usually this function is used in the handler of the event onFormDataSubmit. If the browser supports HTML5 file API formData is instance of FormData, otherwise(like IE10 and older) formData is jQuery representation of the <form> that should be submitted to the server
     *
     * @param {?} formData    If the browser supports HTML5 file API formData is instance of FormData, otherwise(like IE10 and older) formData is jQuery representation of the <form> that should be submitted to the server
     * @param {?} field    Data field that should be appended to the formData. The object has 2 properties - value and name. If the browser supports HTML5 the data field is appended to the formData object. Otherwise it is appended as input hidden field to the <form>
     * @return {?}
     */
    /* istanbul ignore next */
    IgUploadComponent.prototype.addDataField = /**
     * Append additional data field to formData(before submitting it to the server). Usually this function is used in the handler of the event onFormDataSubmit. If the browser supports HTML5 file API formData is instance of FormData, otherwise(like IE10 and older) formData is jQuery representation of the <form> that should be submitted to the server
     *
     * @param {?} formData    If the browser supports HTML5 file API formData is instance of FormData, otherwise(like IE10 and older) formData is jQuery representation of the <form> that should be submitted to the server
     * @param {?} field    Data field that should be appended to the formData. The object has 2 properties - value and name. If the browser supports HTML5 the data field is appended to the formData object. Otherwise it is appended as input hidden field to the <form>
     * @return {?}
     */
    /* istanbul ignore next */
    function (formData, field) { return; };
    ;
    /**
     * Append additional data fields to formData(before submitting it to the server). Usually this function is used in the handler of the event onFormDataSubmit. If the browser supports HTML5 file API formData is instance of FormData, otherwise(like IE10 and older) formData is jQuery representation of the <form> that should be submitted to the server
     *
     * @param formData    If the browser supports HTML5 file API formData is instance of FormData, otherwise(like IE10 and older) formData is jQuery representation of the <form> that should be submitted to the server
     * @param fields    Array of data fields that should be appended to the formData. Each data field is object with 2 properties - value and name. If the browser supports HTML5 these data fields are added to the formData. Otherwise each of these data field is appended as input hidden field to the <form>
     */
    /* istanbul ignore next */
    /**
     * Append additional data fields to formData(before submitting it to the server). Usually this function is used in the handler of the event onFormDataSubmit. If the browser supports HTML5 file API formData is instance of FormData, otherwise(like IE10 and older) formData is jQuery representation of the <form> that should be submitted to the server
     *
     * @param {?} formData    If the browser supports HTML5 file API formData is instance of FormData, otherwise(like IE10 and older) formData is jQuery representation of the <form> that should be submitted to the server
     * @param {?} fields    Array of data fields that should be appended to the formData. Each data field is object with 2 properties - value and name. If the browser supports HTML5 these data fields are added to the formData. Otherwise each of these data field is appended as input hidden field to the <form>
     * @return {?}
     */
    /* istanbul ignore next */
    IgUploadComponent.prototype.addDataFields = /**
     * Append additional data fields to formData(before submitting it to the server). Usually this function is used in the handler of the event onFormDataSubmit. If the browser supports HTML5 file API formData is instance of FormData, otherwise(like IE10 and older) formData is jQuery representation of the <form> that should be submitted to the server
     *
     * @param {?} formData    If the browser supports HTML5 file API formData is instance of FormData, otherwise(like IE10 and older) formData is jQuery representation of the <form> that should be submitted to the server
     * @param {?} fields    Array of data fields that should be appended to the formData. Each data field is object with 2 properties - value and name. If the browser supports HTML5 these data fields are added to the formData. Otherwise each of these data field is appended as input hidden field to the <form>
     * @return {?}
     */
    /* istanbul ignore next */
    function (formData, fields) { return; };
    ;
    /**
     * Start uploading file as submitting form with the specified formNumber.
     *
     * @param formNumber    id of the upload form
     */
    /* istanbul ignore next */
    /**
     * Start uploading file as submitting form with the specified formNumber.
     *
     * @param {?} formNumber    id of the upload form
     * @return {?}
     */
    /* istanbul ignore next */
    IgUploadComponent.prototype.startUpload = /**
     * Start uploading file as submitting form with the specified formNumber.
     *
     * @param {?} formNumber    id of the upload form
     * @return {?}
     */
    /* istanbul ignore next */
    function (formNumber) { return; };
    ;
    /**
     * Cancel upload for the specified file id
     * 			 formNumber - id of the file to be canceled
     *
     * @param formNumber    id of the form which should be cancelled
     */
    /* istanbul ignore next */
    /**
     * Cancel upload for the specified file id
     * 			 formNumber - id of the file to be canceled
     *
     * @param {?} formNumber    id of the form which should be cancelled
     * @return {?}
     */
    /* istanbul ignore next */
    IgUploadComponent.prototype.cancelUpload = /**
     * Cancel upload for the specified file id
     * 			 formNumber - id of the file to be canceled
     *
     * @param {?} formNumber    id of the form which should be cancelled
     * @return {?}
     */
    /* istanbul ignore next */
    function (formNumber) { return; };
    ;
    /**
     * Destroy the widget
     */
    /* istanbul ignore next */
    /**
     * Destroy the widget
     * @return {?}
     */
    /* istanbul ignore next */
    IgUploadComponent.prototype.destroy = /**
     * Destroy the widget
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    /**
     * Returns the information about uploading files - all files uploaded/uploading/pending
     */
    /* istanbul ignore next */
    /**
     * Returns the information about uploading files - all files uploaded/uploading/pending
     * @return {?}
     */
    /* istanbul ignore next */
    IgUploadComponent.prototype.getFileInfoData = /**
     * Returns the information about uploading files - all files uploaded/uploading/pending
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    /**
     * Cancel all uploading and pending files
     */
    /* istanbul ignore next */
    /**
     * Cancel all uploading and pending files
     * @return {?}
     */
    /* istanbul ignore next */
    IgUploadComponent.prototype.cancelAll = /**
     * Cancel all uploading and pending files
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    /**
     * Returns the information about the file by specified file identifier. It could be file which is uploading/uploaded or uploading is not started. If there isn"t file with the specified file id returns null
     *
     * @param fileIndex    unique identifier of the file
     */
    /* istanbul ignore next */
    /**
     * Returns the information about the file by specified file identifier. It could be file which is uploading/uploaded or uploading is not started. If there isn"t file with the specified file id returns null
     *
     * @param {?} fileIndex    unique identifier of the file
     * @return {?}
     */
    /* istanbul ignore next */
    IgUploadComponent.prototype.getFileInfo = /**
     * Returns the information about the file by specified file identifier. It could be file which is uploading/uploaded or uploading is not started. If there isn"t file with the specified file id returns null
     *
     * @param {?} fileIndex    unique identifier of the file
     * @return {?}
     */
    /* istanbul ignore next */
    function (fileIndex) { return; };
    ;
    IgUploadComponent.decorators = [
        { type: Component, args: [{
                    selector: "ig-upload",
                    template: "<ng-content></ng-content>",
                    inputs: ["widgetId", "options", "changeDetectionInterval", "disabled", "create", "width", "height", "autostartupload", "labelUploadButton", "labelAddButton", "labelClearAllButton", "labelSummaryTemplate", "labelSummaryProgressBarTemplate", "labelShowDetails", "labelHideDetails", "labelSummaryProgressButtonCancel", "labelSummaryProgressButtonContinue", "labelSummaryProgressButtonDone", "labelProgressBarFileNameContinue", "errorMessageMaxFileSizeExceeded", "errorMessageGetFileStatus", "errorMessageCancelUpload", "errorMessageNoSuchFile", "errorMessageOther", "errorMessageValidatingFileExtension", "errorMessageAJAXRequestFileSize", "errorMessageTryToRemoveNonExistingFile", "errorMessageTryToStartNonExistingFile", "errorMessageMaxUploadedFiles", "errorMessageMaxSimultaneousFiles", "errorMessageDropMultipleFilesWhenSingleModel", "uploadUrl", "progressUrl", "allowedExtensions", "showFileExtensionIcon", "css", "fileExtensionIcons", "mode", "multipleFiles", "maxUploadedFiles", "maxSimultaneousFilesUploads", "fileSizeMetric", "controlId", "fileSizeDecimalDisplay", "maxFileSize"],
                    outputs: ["fileSelecting", "fileSelected", "fileUploading", "fileUploaded", "fileUploadAborted", "cancelAllClicked", "onError", "fileExtensionsValidating", "onXHRLoad", "onFormDataSubmit"]
                }] }
    ];
    /** @nocollapse */
    IgUploadComponent.ctorParameters = function () { return [
        { type: ElementRef },
        { type: Renderer },
        { type: IterableDiffers },
        { type: KeyValueDiffers },
        { type: ChangeDetectorRef }
    ]; };
    return IgUploadComponent;
}(IgControlBase));
export { IgUploadComponent };
