// igniteui-angular-wrappers/esm5/lib/igscheduler/igscheduler.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Component, ElementRef, Renderer, IterableDiffers, KeyValueDiffers, ChangeDetectorRef } from "@angular/core";
import { IgControlBase } from "../igcontrolbase/igcontrolbase";
var IgSchedulerComponent = /** @class */ (function (_super) {
    tslib_1.__extends(IgSchedulerComponent, _super);
    function IgSchedulerComponent(el, renderer, differs, kvalDiffers, cdr) {
        return _super.call(this, el, renderer, differs, kvalDiffers, cdr) || this;
    }
    /**
     * Gets reference to appointment by id
     *
     * @param id
     */
    /* istanbul ignore next */
    /**
     * Gets reference to appointment by id
     *
     * @param {?} id
     * @return {?}
     */
    /* istanbul ignore next */
    IgSchedulerComponent.prototype.getAppointmentById = /**
     * Gets reference to appointment by id
     *
     * @param {?} id
     * @return {?}
     */
    /* istanbul ignore next */
    function (id) { return; };
    ;
    /**
     * Creates a new appointment and renders it to the scheduler
     *
     * @param appointment
     */
    /* istanbul ignore next */
    /**
     * Creates a new appointment and renders it to the scheduler
     *
     * @param {?} appointment
     * @return {?}
     */
    /* istanbul ignore next */
    IgSchedulerComponent.prototype.createAppointment = /**
     * Creates a new appointment and renders it to the scheduler
     *
     * @param {?} appointment
     * @return {?}
     */
    /* istanbul ignore next */
    function (appointment) { return; };
    ;
    /**
     * Deletes appointment from the appointment collection
     *
     * @param appointment	appointment
     */
    /* istanbul ignore next */
    /**
     * Deletes appointment from the appointment collection
     *
     * @param {?} appointment 	appointment
     * @return {?}
     */
    /* istanbul ignore next */
    IgSchedulerComponent.prototype.deleteAppointment = /**
     * Deletes appointment from the appointment collection
     *
     * @param {?} appointment 	appointment
     * @return {?}
     */
    /* istanbul ignore next */
    function (appointment) { return; };
    ;
    /**
     * Deletes appointment from the appointment collection
     *
     * @param appointment	appointment
     * @param updateAppoinment	updateAppoinment
     */
    /* istanbul ignore next */
    /**
     * Deletes appointment from the appointment collection
     *
     * @param {?} appointment 	appointment
     * @param {?} updateAppoinment 	updateAppoinment
     * @return {?}
     */
    /* istanbul ignore next */
    IgSchedulerComponent.prototype.editAppointment = /**
     * Deletes appointment from the appointment collection
     *
     * @param {?} appointment 	appointment
     * @param {?} updateAppoinment 	updateAppoinment
     * @return {?}
     */
    /* istanbul ignore next */
    function (appointment, updateAppoinment) { return; };
    ;
    /**
     * Destroys the widget
     */
    /* istanbul ignore next */
    /**
     * Destroys the widget
     * @return {?}
     */
    /* istanbul ignore next */
    IgSchedulerComponent.prototype.destroy = /**
     * Destroys the widget
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    /**
     * Gets reference to the today UI button.
     */
    /* istanbul ignore next */
    /**
     * Gets reference to the today UI button.
     * @return {?}
     */
    /* istanbul ignore next */
    IgSchedulerComponent.prototype.todayButton = /**
     * Gets reference to the today UI button.
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    /**
     * Gets reference to the previous UI button.
     */
    /* istanbul ignore next */
    /**
     * Gets reference to the previous UI button.
     * @return {?}
     */
    /* istanbul ignore next */
    IgSchedulerComponent.prototype.previousButton = /**
     * Gets reference to the previous UI button.
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    /**
     * Gets reference to the date range UI button.
     */
    /* istanbul ignore next */
    /**
     * Gets reference to the date range UI button.
     * @return {?}
     */
    /* istanbul ignore next */
    IgSchedulerComponent.prototype.dateRangeButton = /**
     * Gets reference to the date range UI button.
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    /**
     * Gets reference to the next UI button.
     */
    /* istanbul ignore next */
    /**
     * Gets reference to the next UI button.
     * @return {?}
     */
    /* istanbul ignore next */
    IgSchedulerComponent.prototype.nextButton = /**
     * Gets reference to the next UI button.
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    /**
     * Gets reference to the jQuery calendar UI control.
     */
    /* istanbul ignore next */
    /**
     * Gets reference to the jQuery calendar UI control.
     * @return {?}
     */
    /* istanbul ignore next */
    IgSchedulerComponent.prototype.getCalendar = /**
     * Gets reference to the jQuery calendar UI control.
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    IgSchedulerComponent.decorators = [
        { type: Component, args: [{
                    selector: "ig-scheduler",
                    template: "<ng-content></ng-content>",
                    inputs: ["widgetId", "options", "changeDetectionInterval", "disabled", "create", "views", "viewMode", "selectedDate", "enableTodayButton", "width", "height", "agendaViewSettings", "monthViewSettings", "appointmentDialogSuppress"],
                    outputs: ["agendaRangeChanging", "agendaRangeChanged", "daySelected", "monthChanging", "monthChanged", "rendering", "rendered", "viewChanging", "viewChanged", "appointmentDialogOpening", "appointmentDialogOpened", "appointmentDialogClosing", "appointmentDialogClosed", "appointmentCreating", "appointmentCreated", "appointmentDeleting", "appointmentDeleted", "appointmentEditing", "appointmentEdited"]
                }] }
    ];
    /** @nocollapse */
    IgSchedulerComponent.ctorParameters = function () { return [
        { type: ElementRef },
        { type: Renderer },
        { type: IterableDiffers },
        { type: KeyValueDiffers },
        { type: ChangeDetectorRef }
    ]; };
    return IgSchedulerComponent;
}(IgControlBase));
export { IgSchedulerComponent };
