// igniteui-angular-wrappers/esm5/lib/igqrcodebarcode/igqrcodebarcode.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Component, ElementRef, Renderer, IterableDiffers, KeyValueDiffers, ChangeDetectorRef } from "@angular/core";
import { IgControlBase } from "../igcontrolbase/igcontrolbase";
var IgQRCodeBarcodeComponent = /** @class */ (function (_super) {
    tslib_1.__extends(IgQRCodeBarcodeComponent, _super);
    function IgQRCodeBarcodeComponent(el, renderer, differs, kvalDiffers, cdr) {
        return _super.call(this, el, renderer, differs, kvalDiffers, cdr) || this;
    }
    /**
     * @return {?}
     */
    IgQRCodeBarcodeComponent.prototype.option = /**
     * @return {?}
     */
    function () { return; };
    ;
    /**
     * Returns information about how the barcode is rendered.
     */
    /* istanbul ignore next */
    /**
     * Returns information about how the barcode is rendered.
     * @return {?}
     */
    /* istanbul ignore next */
    IgQRCodeBarcodeComponent.prototype.exportVisualData = /**
     * Returns information about how the barcode is rendered.
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    /**
     * Causes all pending changes of the barcode e.g. by changed property values to be rendered immediately.
     */
    /* istanbul ignore next */
    /**
     * Causes all pending changes of the barcode e.g. by changed property values to be rendered immediately.
     * @return {?}
     */
    /* istanbul ignore next */
    IgQRCodeBarcodeComponent.prototype.flush = /**
     * Causes all pending changes of the barcode e.g. by changed property values to be rendered immediately.
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    /**
     * Destroys widget.
     */
    /* istanbul ignore next */
    /**
     * Destroys widget.
     * @return {?}
     */
    /* istanbul ignore next */
    IgQRCodeBarcodeComponent.prototype.destroy = /**
     * Destroys widget.
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    /**
     * Re-polls the css styles for the widget. Use this method when the css styles have been modified.
     */
    /* istanbul ignore next */
    /**
     * Re-polls the css styles for the widget. Use this method when the css styles have been modified.
     * @return {?}
     */
    /* istanbul ignore next */
    IgQRCodeBarcodeComponent.prototype.styleUpdated = /**
     * Re-polls the css styles for the widget. Use this method when the css styles have been modified.
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    IgQRCodeBarcodeComponent.decorators = [
        { type: Component, args: [{
                    selector: "ig-q-r-code-barcode",
                    template: "<ng-content></ng-content>",
                    inputs: ["widgetId", "options", "changeDetectionInterval", "disabled", "create", "width", "height", "backingBrush", "backingOutline", "backingStrokeThickness", "barBrush", "fontBrush", "font", "data", "errorMessageText", "stretch", "barsFillMode", "widthToHeightRatio", "xDimension", "errorCorrectionLevel", "sizeVersion", "encodingMode", "eciNumber", "eciHeaderDisplayMode", "fnc1Mode", "applicationIndicator"],
                    outputs: ["errorMessageDisplaying", "dataChanged"]
                }] }
    ];
    /** @nocollapse */
    IgQRCodeBarcodeComponent.ctorParameters = function () { return [
        { type: ElementRef },
        { type: Renderer },
        { type: IterableDiffers },
        { type: KeyValueDiffers },
        { type: ChangeDetectorRef }
    ]; };
    return IgQRCodeBarcodeComponent;
}(IgControlBase));
export { IgQRCodeBarcodeComponent };
