// igniteui-angular-wrappers/esm5/lib/igsplitter/igsplitter.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { ElementRef, Renderer, IterableDiffers, Component, KeyValueDiffers, ChangeDetectorRef } from "@angular/core";
import { IgContentControlBase } from "../igcontrolbase/igcontentcontrolbase";
var IgSplitterComponent = /** @class */ (function (_super) {
    tslib_1.__extends(IgSplitterComponent, _super);
    function IgSplitterComponent(el, renderer, differs, kvalDiffers, cdr) {
        return _super.call(this, el, renderer, differs, kvalDiffers, cdr) || this;
    }
    /**
     * Returns the element that represents this widget.
     */
    /* istanbul ignore next */
    /**
     * Returns the element that represents this widget.
     * @return {?}
     */
    /* istanbul ignore next */
    IgSplitterComponent.prototype.widget = /**
     * Returns the element that represents this widget.
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    /**
     * Expand the specified panel by index.
     *
     * @param index Specifies the index of the panel to expand.
     */
    /* istanbul ignore next */
    /**
     * Expand the specified panel by index.
     *
     * @param {?} index Specifies the index of the panel to expand.
     * @return {?}
     */
    /* istanbul ignore next */
    IgSplitterComponent.prototype.expandAt = /**
     * Expand the specified panel by index.
     *
     * @param {?} index Specifies the index of the panel to expand.
     * @return {?}
     */
    /* istanbul ignore next */
    function (index) { return; };
    ;
    /**
     * Collapse the specified panel.
     *
     * @param index Specifies the index of the panel to collapse.
     */
    /* istanbul ignore next */
    /**
     * Collapse the specified panel.
     *
     * @param {?} index Specifies the index of the panel to collapse.
     * @return {?}
     */
    /* istanbul ignore next */
    IgSplitterComponent.prototype.collapseAt = /**
     * Collapse the specified panel.
     *
     * @param {?} index Specifies the index of the panel to collapse.
     * @return {?}
     */
    /* istanbul ignore next */
    function (index) { return; };
    ;
    /**
     * Retrieves the jQuery element of the first panel.
     */
    /* istanbul ignore next */
    /**
     * Retrieves the jQuery element of the first panel.
     * @return {?}
     */
    /* istanbul ignore next */
    IgSplitterComponent.prototype.firstPanel = /**
     * Retrieves the jQuery element of the first panel.
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    /**
     * Retrieves the jQuery element of the second panel.
     */
    /* istanbul ignore next */
    /**
     * Retrieves the jQuery element of the second panel.
     * @return {?}
     */
    /* istanbul ignore next */
    IgSplitterComponent.prototype.secondPanel = /**
     * Retrieves the jQuery element of the second panel.
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    /**
     * Refresh splitter layout, use this method to re-render the splitter if some changes to the layout are applied.
     */
    /* istanbul ignore next */
    /**
     * Refresh splitter layout, use this method to re-render the splitter if some changes to the layout are applied.
     * @return {?}
     */
    /* istanbul ignore next */
    IgSplitterComponent.prototype.refreshLayout = /**
     * Refresh splitter layout, use this method to re-render the splitter if some changes to the layout are applied.
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    /**
     * You can set new size of the first panel after the splitter is rendered.
     *
     * @param size Specifies the new size of the first panel.
     */
    /* istanbul ignore next */
    /**
     * You can set new size of the first panel after the splitter is rendered.
     *
     * @param {?} size Specifies the new size of the first panel.
     * @return {?}
     */
    /* istanbul ignore next */
    IgSplitterComponent.prototype.setFirstPanelSize = /**
     * You can set new size of the first panel after the splitter is rendered.
     *
     * @param {?} size Specifies the new size of the first panel.
     * @return {?}
     */
    /* istanbul ignore next */
    function (size) { return; };
    ;
    /**
     * You can set new size of the second panel after the splitter is rendered.
     *
     * @param size Specifies the new size of the second panel.
     */
    /* istanbul ignore next */
    /**
     * You can set new size of the second panel after the splitter is rendered.
     *
     * @param {?} size Specifies the new size of the second panel.
     * @return {?}
     */
    /* istanbul ignore next */
    IgSplitterComponent.prototype.setSecondPanelSize = /**
     * You can set new size of the second panel after the splitter is rendered.
     *
     * @param {?} size Specifies the new size of the second panel.
     * @return {?}
     */
    /* istanbul ignore next */
    function (size) { return; };
    ;
    /**
     * Destroys the igSplitter widget
     */
    /* istanbul ignore next */
    /**
     * Destroys the igSplitter widget
     * @return {?}
     */
    /* istanbul ignore next */
    IgSplitterComponent.prototype.destroy = /**
     * Destroys the igSplitter widget
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    IgSplitterComponent.decorators = [
        { type: Component, args: [{
                    selector: "ig-splitter",
                    template: "<ng-content></ng-content>",
                    inputs: ["widgetId", "options", "changeDetectionInterval", "disabled", "create", "width", "height", "orientation", "panels", "dragDelta", "resizeOtherSplitters"],
                    outputs: ["collapsed", "expanded", "resizeStarted", "resizing", "resizeEnded", "layoutRefreshing", "layoutRefreshed"]
                }] }
    ];
    /** @nocollapse */
    IgSplitterComponent.ctorParameters = function () { return [
        { type: ElementRef },
        { type: Renderer },
        { type: IterableDiffers },
        { type: KeyValueDiffers },
        { type: ChangeDetectorRef }
    ]; };
    return IgSplitterComponent;
}(IgContentControlBase));
export { IgSplitterComponent };
