// igniteui-angular-wrappers/esm5/lib/igtreegrid/igtreegrid.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Component, Renderer, ElementRef, IterableDiffers, KeyValueDiffers, ChangeDetectorRef } from "@angular/core";
import { IgGridBase } from "../iggrid/iggridbase";
var IgTreeGridComponent = /** @class */ (function (_super) {
    tslib_1.__extends(IgTreeGridComponent, _super);
    function IgTreeGridComponent(el, renderer, differs, kvalDiffers, cdr) {
        return _super.call(this, el, renderer, differs, kvalDiffers, cdr) || this;
    }
    /**
     * @param {?} id
     * @return {?}
     */
    IgTreeGridComponent.prototype.deleteRow = /**
     * @param {?} id
     * @return {?}
     */
    function (id) {
        /** @type {?} */
        var element = jQuery(this._el);
        /** @type {?} */
        var tr = element.find("tr[data-id='" + id + "']");
        /** @type {?} */
        var dataLevel = tr.attr("aria-level");
        if (tr.length > 0) {
            element.data(this._widgetName).dataSource.deleteRow(id, true);
            element.data(this._widgetName).dataSource._removeTransactionsByRecordId(id);
            /** @type {?} */
            var trs = tr.nextUntil("tr[data-level=" + dataLevel + "]");
            if (trs.length === 0) {
                trs = tr.nextAll("tr[data-level]");
            }
            tr.remove();
            trs.remove();
        }
    };
    /**
     * @param {?} rec
     * @param {?} currValue
     * @param {?} key
     * @return {?}
     */
    IgTreeGridComponent.prototype.updateRow = /**
     * @param {?} rec
     * @param {?} currValue
     * @param {?} key
     * @return {?}
     */
    function (rec, currValue, key) {
        /** @type {?} */
        var element = jQuery(this._el);
        /** @type {?} */
        var grid = element.data(this._widgetName);
        /** @type {?} */
        var childDataKey = this["childDataKey"] || this.options.childDataKey;
        /** @type {?} */
        var column = element.data(this._widgetName).columnByKey(key);
        if (!column && key === childDataKey) {
            //we have an hierarchical data source and one of the nested collections has changed.
            grid.dataBind();
        }
        else {
            _super.prototype.updateRow.call(this, rec, currValue, key);
        }
    };
    /**
     * @return {?}
     */
    IgTreeGridComponent.prototype.markForCheck = /**
     * @return {?}
     */
    function () {
        _super.prototype.markForCheck.call(this);
        /** @type {?} */
        var childDataKey = this["childDataKey"] || this.options.childDataKey;
        if (childDataKey) {
            this.dataBind();
        }
    };
    /**
     * Clears the transaction log (delegates to igDataSource). Note that this does not update the UI. In case the UI must be updated, set the second parameter "updateUI" to true, which will trigger a call to dataBind() to re-render the contents.
     *
     * @param rowId     If specified, will only rollback the transactions with that row id.
     * @param updateUI     Whether to update the UI or not.
     */
    /* istanbul ignore next */
    /**
     * Clears the transaction log (delegates to igDataSource). Note that this does not update the UI. In case the UI must be updated, set the second parameter "updateUI" to true, which will trigger a call to dataBind() to re-render the contents.
     *
     * @param {?=} rowId     If specified, will only rollback the transactions with that row id.
     * @param {?=} updateUI     Whether to update the UI or not.
     * @return {?}
     */
    /* istanbul ignore next */
    IgTreeGridComponent.prototype.rollback = /**
     * Clears the transaction log (delegates to igDataSource). Note that this does not update the UI. In case the UI must be updated, set the second parameter "updateUI" to true, which will trigger a call to dataBind() to re-render the contents.
     *
     * @param {?=} rowId     If specified, will only rollback the transactions with that row id.
     * @param {?=} updateUI     Whether to update the UI or not.
     * @return {?}
     */
    /* istanbul ignore next */
    function (rowId, updateUI) { return; };
    ;
    /**
     * Causes the treegrid to data bind to the data source (local or remote) , and re-render all of the data
     */
    /* istanbul ignore next */
    /**
     * Causes the treegrid to data bind to the data source (local or remote) , and re-render all of the data
     * @return {?}
     */
    /* istanbul ignore next */
    IgTreeGridComponent.prototype.dataBind = /**
     * Causes the treegrid to data bind to the data source (local or remote) , and re-render all of the data
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    /**
     * Toggle row by specified row or row identifier
     *
     * @param row     jQuery table row object or a row id.
     * @param callback     Specifies a custom function to be called when row is expanded/collapsed. The callback has 4 arguments- a reference to the current context(this), object that holds 2 properties(unfixedRow - DOM representation of the unfixed row, fixedRow - DOM representation of the fixed row, if there is no fixed columns it is undefined), reference to the dataRecord, expand - specifies whether row is expanded
     */
    /* istanbul ignore next */
    /**
     * Toggle row by specified row or row identifier
     *
     * @param {?} row     jQuery table row object or a row id.
     * @param {?=} callback     Specifies a custom function to be called when row is expanded/collapsed. The callback has 4 arguments- a reference to the current context(this), object that holds 2 properties(unfixedRow - DOM representation of the unfixed row, fixedRow - DOM representation of the fixed row, if there is no fixed columns it is undefined), reference to the dataRecord, expand - specifies whether row is expanded
     * @return {?}
     */
    /* istanbul ignore next */
    IgTreeGridComponent.prototype.toggleRow = /**
     * Toggle row by specified row or row identifier
     *
     * @param {?} row     jQuery table row object or a row id.
     * @param {?=} callback     Specifies a custom function to be called when row is expanded/collapsed. The callback has 4 arguments- a reference to the current context(this), object that holds 2 properties(unfixedRow - DOM representation of the unfixed row, fixedRow - DOM representation of the fixed row, if there is no fixed columns it is undefined), reference to the dataRecord, expand - specifies whether row is expanded
     * @return {?}
     */
    /* istanbul ignore next */
    function (row, callback) { return; };
    ;
    /**
     * Expands a parent row by specified row or row identifier
     *
     * @param row     jQuery table row object or a row id.
     * @param callback     Specifies a custom function to be called when row is expanded/collapsed. The callback has 4 arguments- a reference to the current context(this), object that holds 2 properties(unfixedRow - DOM representation of the unfixed row, fixedRow - DOM representation of the fixed row, if there is no fixed columns it is undefined), reference to the dataRecord, expand - specifies whether row is expanded
     */
    /* istanbul ignore next */
    /**
     * Expands a parent row by specified row or row identifier
     *
     * @param {?} row     jQuery table row object or a row id.
     * @param {?=} callback     Specifies a custom function to be called when row is expanded/collapsed. The callback has 4 arguments- a reference to the current context(this), object that holds 2 properties(unfixedRow - DOM representation of the unfixed row, fixedRow - DOM representation of the fixed row, if there is no fixed columns it is undefined), reference to the dataRecord, expand - specifies whether row is expanded
     * @return {?}
     */
    /* istanbul ignore next */
    IgTreeGridComponent.prototype.expandRow = /**
     * Expands a parent row by specified row or row identifier
     *
     * @param {?} row     jQuery table row object or a row id.
     * @param {?=} callback     Specifies a custom function to be called when row is expanded/collapsed. The callback has 4 arguments- a reference to the current context(this), object that holds 2 properties(unfixedRow - DOM representation of the unfixed row, fixedRow - DOM representation of the fixed row, if there is no fixed columns it is undefined), reference to the dataRecord, expand - specifies whether row is expanded
     * @return {?}
     */
    /* istanbul ignore next */
    function (row, callback) { return; };
    ;
    /**
     * Collapses a parent row by specified row or row identifier
     *
     * @param row     jQuery table row object, raw DOM row object or a row id.
     * @param callback     Specifies a custom function to be called when row is expanded/collapsed. The callback has 4 arguments- a reference to the current context(this), object that holds 2 properties(unfixedRow - DOM representation of the unfixed row, fixedRow - DOM representation of the fixed row, if there is no fixed columns it is undefined), reference to the dataRecord, expand - specifies whether row is expanded
     */
    /* istanbul ignore next */
    /**
     * Collapses a parent row by specified row or row identifier
     *
     * @param {?} row     jQuery table row object, raw DOM row object or a row id.
     * @param {?=} callback     Specifies a custom function to be called when row is expanded/collapsed. The callback has 4 arguments- a reference to the current context(this), object that holds 2 properties(unfixedRow - DOM representation of the unfixed row, fixedRow - DOM representation of the fixed row, if there is no fixed columns it is undefined), reference to the dataRecord, expand - specifies whether row is expanded
     * @return {?}
     */
    /* istanbul ignore next */
    IgTreeGridComponent.prototype.collapseRow = /**
     * Collapses a parent row by specified row or row identifier
     *
     * @param {?} row     jQuery table row object, raw DOM row object or a row id.
     * @param {?=} callback     Specifies a custom function to be called when row is expanded/collapsed. The callback has 4 arguments- a reference to the current context(this), object that holds 2 properties(unfixedRow - DOM representation of the unfixed row, fixedRow - DOM representation of the fixed row, if there is no fixed columns it is undefined), reference to the dataRecord, expand - specifies whether row is expanded
     * @return {?}
     */
    /* istanbul ignore next */
    function (row, callback) { return; };
    ;
    /**
     * Adds a new row (TR) to the grid as a child of a specific row, by taking a data row object. Assumes the record will have the primary key.
     *
     * @param rec     The data row JavaScript object.
     * @param parentId     Identifier/key of the targeted parent row. If missing, then the new row is rendered to the bottom of the grid.
     */
    /* istanbul ignore next */
    /**
     * Adds a new row (TR) to the grid as a child of a specific row, by taking a data row object. Assumes the record will have the primary key.
     *
     * @param {?} rec     The data row JavaScript object.
     * @param {?=} parentId     Identifier/key of the targeted parent row. If missing, then the new row is rendered to the bottom of the grid.
     * @return {?}
     */
    /* istanbul ignore next */
    IgTreeGridComponent.prototype.renderNewChild = /**
     * Adds a new row (TR) to the grid as a child of a specific row, by taking a data row object. Assumes the record will have the primary key.
     *
     * @param {?} rec     The data row JavaScript object.
     * @param {?=} parentId     Identifier/key of the targeted parent row. If missing, then the new row is rendered to the bottom of the grid.
     * @return {?}
     */
    /* istanbul ignore next */
    function (rec, parentId) { return; };
    ;
    /**
     * Destroys igTreeGrid
     */
    /* istanbul ignore next */
    /**
     * Destroys igTreeGrid
     * @return {?}
     */
    /* istanbul ignore next */
    IgTreeGridComponent.prototype.destroy = /**
     * Destroys igTreeGrid
     * @return {?}
     */
    /* istanbul ignore next */
    function () { return; };
    ;
    IgTreeGridComponent.decorators = [
        { type: Component, args: [{
                    selector: "ig-tree-grid",
                    template: "<ng-content></ng-content>",
                    inputs: ["widgetId", "options", "changeDetectionInterval", "disabled", "create", "width", "height", "autoAdjustHeight", "avgRowHeight", "avgColumnWidth", "defaultColumnWidth", "autoGenerateColumns", "virtualization", "virtualizationMode", "requiresDataBinding", "rowVirtualization", "columnVirtualization", "virtualizationMouseWheelStep", "adjustVirtualHeights", "templatingEngine", "columns", "dataSource", "dataSourceUrl", "dataSourceType", "responseDataKey", "responseTotalRecCountKey", "requestType", "responseContentType", "showHeader", "showFooter", "fixedHeaders", "fixedFooters", "caption", "features", "tabIndex", "localSchemaTransform", "primaryKey", "serializeTransactionLog", "autoCommit", "aggregateTransactions", "autoFormat", "renderCheckboxes", "updateUrl", "restSettings", "alternateRowStyles", "autofitLastColumn", "enableHoverStyles", "enableUTCDates", "mergeUnboundColumns", "jsonpRequest", "enableResizeContainerCheck", "featureChooserIconDisplay", "scrollSettings", "indentation", "initialIndentationLevel", "showExpansionIndicator", "expandTooltipText", "collapseTooltipText", "foreignKey", "initialExpandDepth", "foreignKeyRootValue", "renderExpansionIndicatorColumn", "renderFirstDataCellFunction", "childDataKey", "renderExpansionCellFunction", "enableRemoteLoadOnDemand", "dataSourceSettings"],
                    outputs: ["cellClick", "cellRightClick", "dataBinding", "dataBound", "rendering", "rendered", "dataRendering", "dataRendered", "headerRendering", "headerRendered", "footerRendering", "footerRendered", "headerCellRendered", "rowsRendering", "rowsRendered", "schemaGenerated", "columnsCollectionModified", "requestError", "created", "destroyed", "rowExpanding", "rowExpanded", "rowCollapsing", "rowCollapsed"]
                }] }
    ];
    /** @nocollapse */
    IgTreeGridComponent.ctorParameters = function () { return [
        { type: ElementRef },
        { type: Renderer },
        { type: IterableDiffers },
        { type: KeyValueDiffers },
        { type: ChangeDetectorRef }
    ]; };
    return IgTreeGridComponent;
}(IgGridBase));
export { IgTreeGridComponent };
